diff --git a/CMakeLists.txt b/CMakeLists.txt
index b35d92b..b49574a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -118,11 +118,11 @@ if ( LUA_BUILD_AS_DLL )
   set_target_properties ( liblua PROPERTIES COMPILE_DEFINITIONS LUA_BUILD_AS_DLL )
 endif ()
 
-add_executable ( lua ${SRC_LUA} src/lua.rc )
-target_link_libraries ( lua liblua )
+#add_executable ( lua ${SRC_LUA} src/lua.rc )
+#target_link_libraries ( lua liblua )
 
-add_executable ( luac ${SRC_CORE} ${SRC_LIB} ${SRC_LUAC} src/luac.rc )
-target_link_libraries ( luac ${LIBS} )
+#add_executable ( luac ${SRC_CORE} ${SRC_LIB} ${SRC_LUAC} src/luac.rc )
+#target_link_libraries ( luac ${LIBS} )
 
 # On windows a variant of the lua interpreter without console output needs to be built
 if ( LUA_BUILD_WLUA )
@@ -131,11 +131,11 @@ if ( LUA_BUILD_WLUA )
   install_executable ( wlua )
 endif ( )
 
-install_executable ( lua luac )
+#install_executable ( lua luac )
 install_library ( liblua )
-install_data ( README.md )
+#install_data ( README.md )
 #install_lua_module ( strict etc/strict.lua )
 install_header ( src/lua.h src/lualib.h src/lauxlib.h src/lua.hpp ${CMAKE_CURRENT_BINARY_DIR}/luaconf.h )
-install_doc ( doc/ )
-install_foo ( etc/ )
+#install_doc ( doc/ )
+#install_foo ( etc/ )
 #install_test ( test/ )
