diff --git a/src/mod_maxminddb.c b/src/mod_maxminddb.c
index 01f32f8..0927163 100644
--- a/src/mod_maxminddb.c
+++ b/src/mod_maxminddb.c
@@ -233,6 +233,8 @@ static apr_status_t cleanup_database(void *mmdb)
     return APR_SUCCESS;
 }
 
+#define max_path_segments 80
+
 static const char *set_maxminddb_env(cmd_parms *cmd, void *dir_config,
                                      const char *env, const char *path)
 {
@@ -240,7 +242,7 @@ static const char *set_maxminddb_env(cmd_parms *cmd, void *dir_config,
 
     INFO(cmd->server, "set_maxminddb_env (server) %s %s", env, path);
 
-    const int max_path_segments = 80;
+    // const int max_path_segments = 80;
     char *path_segments[max_path_segments];
 
     char *tokenized_path = apr_pstrdup(cmd->pool, path);
