diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6d57d58b..2c3bdeb1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,13 +26,19 @@ OPTION(ENABLE_ASAN "Enable (gcc) ASAN support" 0)
 OPTION(ENABLE_CPP "Enable CPP GD API" 1)
 OPTION(VERBOSE_MAKEFILE "Enable CMAKE_VERBOSE_MAKEFILE" 0)
 
+OPTION(BUILD_TESTS "Build tests" 0)
+OPTION(BUILD_EXEMPLES "Build exemples" 0)
+OPTION(BUILD_PROGRAMS "Build programs" 0)
+OPTION(BUILD_DOCS "Build docs" 0)
+OPTION(BUILD_CPACK "Build CPACK" 0)
+
 if (VERBOSE_MAKEFILE)
 	set(CMAKE_VERBOSE_MAKEFILE ON)
 endif (VERBOSE_MAKEFILE)
 
-if (BUILD_TEST)
+if (BUILD_TESTS)
 	ENABLE_TESTING()
-endif(BUILD_TEST)
+endif(BUILD_TESTS)
 
 # if you would like to pass C flags to the compiler, then just
 # - specify -DCMAKE_C_FLAGS=... on invocation of cmake, or
@@ -286,7 +292,7 @@ else (USE_EXT_GD)
 
 	if (WIN32)
 		SET(GD_LIB libgd)
-		ADD_DEFINITIONS( -DWIN32 -D_WIN32 -DMSWIN32 -DBGDWIN32 -DWINVER=0x0500  -D_WIN32_WINNT=0x0500 -D_WIN32_IE=0x0600)
+		ADD_DEFINITIONS( -DWIN32 -D_WIN32 -DMSWIN32 -DBGDWIN32 -DWINVER=0x0A00  -D_WIN32_WINNT=0x0A00 -D_WIN32_IE=0x0600)
 
 		if(NOT MINGW AND MSVC_VERSION GREATER 1399)
 			ADD_DEFINITIONS("/D_CRT_SECURE_NO_DEPRECATE")
@@ -312,59 +318,64 @@ else (USE_EXT_GD)
 	add_subdirectory(src)
 endif (USE_EXT_GD)
 
-add_subdirectory(tests)
-add_subdirectory(examples)
-add_subdirectory(docs)
-
 add_custom_target(distclean ${GD_SOURCE_DIR}/cmake/distclean.sh)
 
-SET(CPACK_PACKAGE_NAME "libgd")
-SET(CPACK_PACKAGE_DESCRIPTION_SUMMARY "libGD, powerful and easy to use graphic library")
-SET(CPACK_PACKAGE_VENDOR "https://www.libgd.org")
-SET(CPACK_PACKAGE_VERSION_MAJOR ${GDLIB_MAJOR})
-SET(CPACK_PACKAGE_VERSION_MINOR ${GDLIB_MINOR})
-SET(CPACK_PACKAGE_VERSION_PATCH ${GDLIB_REVISION}${GDLIB_EXTRA})
-
-if(WIN32)
-	set(CPACK_GENERATOR ZIP)
-else(WIN32)
-	set(CPACK_GENERATOR TGZ)
-endif(WIN32)
-
-
-INSTALL(FILES docs/INSTALL DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
-INSTALL(FILES docs/README.JPN DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
-INSTALL(FILES docs/README.CMAKE DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
-INSTALL(FILES docs/README.TESTING DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
-INSTALL(FILES docs/README.TXT DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
-
-
-INSTALL(FILES examples/arc.c DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
-INSTALL(FILES examples/copyrotated.c DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
-INSTALL(FILES examples/crop.c DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
-INSTALL(FILES examples/flip.c DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
-INSTALL(FILES examples/gif.c DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
-INSTALL(FILES examples/nnquant.c DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
-INSTALL(FILES examples/noIcon.pic DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
-INSTALL(FILES examples/noIcon.sgi DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
-INSTALL(FILES examples/noIcon.tga DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
-INSTALL(FILES examples/noIconAlpha.tga DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
-INSTALL(FILES examples/test_crop_threshold.png DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
-INSTALL(FILES examples/tgaread.c DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
-INSTALL(FILES examples/tiffread.c DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
-INSTALL(FILES examples/windows.c DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
-
-
-set(CPACK_RESOURCE_FILE_LICENSE "${CMAKE_SOURCE_DIR}/COPYING")
-
-set(CPACK_SOURCE_GENERATOR TGZ)
-set(CPACK_SOURCE_IGNORE_FILES
-"~$"
-"\\\\.swp$"
-"\\\\.gitignore$"
-"^${PROJECT_SOURCE_DIR}/debian/"
-"^${PROJECT_SOURCE_DIR}/old/"
-"^${PROJECT_SOURCE_DIR}/bld/"
-)
-install(FILES ${top_level_DOCFILES} DESTINATION ${DOC_DIR})
-INCLUDE(CPack)
+if(BUILD_TESTS)
+	add_subdirectory(tests)
+endif (BUILD_TESTS)
+
+if(BUILD_EXEMPLES)
+	add_subdirectory(examples)
+	INSTALL(FILES examples/arc.c DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
+	INSTALL(FILES examples/copyrotated.c DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
+	INSTALL(FILES examples/crop.c DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
+	INSTALL(FILES examples/flip.c DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
+	INSTALL(FILES examples/gif.c DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
+	INSTALL(FILES examples/nnquant.c DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
+	INSTALL(FILES examples/noIcon.pic DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
+	INSTALL(FILES examples/noIcon.sgi DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
+	INSTALL(FILES examples/noIcon.tga DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
+	INSTALL(FILES examples/noIconAlpha.tga DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
+	INSTALL(FILES examples/test_crop_threshold.png DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
+	INSTALL(FILES examples/tgaread.c DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
+	INSTALL(FILES examples/tiffread.c DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
+	INSTALL(FILES examples/windows.c DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
+endif(BUILD_EXEMPLES)
+
+if(BUILD_DOCS)
+	add_subdirectory(docs)
+	INSTALL(FILES docs/INSTALL DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
+	INSTALL(FILES docs/README.JPN DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
+	INSTALL(FILES docs/README.CMAKE DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
+	INSTALL(FILES docs/README.TESTING DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
+	INSTALL(FILES docs/README.TXT DESTINATION share/doc/gd-${GDLIB_MAJOR}.${GDLIB_MINOR})
+endif(BUILD_DOCS)
+
+if(BUILD_CPACK)
+	SET(CPACK_PACKAGE_NAME "libgd")
+	SET(CPACK_PACKAGE_DESCRIPTION_SUMMARY "libGD, powerful and easy to use graphic library")
+	SET(CPACK_PACKAGE_VENDOR "https://www.libgd.org")
+	SET(CPACK_PACKAGE_VERSION_MAJOR ${GDLIB_MAJOR})
+	SET(CPACK_PACKAGE_VERSION_MINOR ${GDLIB_MINOR})
+	SET(CPACK_PACKAGE_VERSION_PATCH ${GDLIB_REVISION}${GDLIB_EXTRA})
+
+	if(WIN32)
+		set(CPACK_GENERATOR ZIP)
+	else(WIN32)
+		set(CPACK_GENERATOR TGZ)
+	endif(WIN32)
+
+	set(CPACK_RESOURCE_FILE_LICENSE "${CMAKE_SOURCE_DIR}/COPYING")
+
+	set(CPACK_SOURCE_GENERATOR TGZ)
+	set(CPACK_SOURCE_IGNORE_FILES
+	"~$"
+	"\\\\.swp$"
+	"\\\\.gitignore$"
+	"^${PROJECT_SOURCE_DIR}/debian/"
+	"^${PROJECT_SOURCE_DIR}/old/"
+	"^${PROJECT_SOURCE_DIR}/bld/"
+	)
+	install(FILES ${top_level_DOCFILES} DESTINATION ${DOC_DIR})
+	INCLUDE(CPack)
+endif(BUILD_CPACK)
\ No newline at end of file
diff --git "a/src/CMakeLists.txt" "b/src/CMakeLists.txt"
index e24dc014..a4eba7d1 100644
--- "a/src/CMakeLists.txt"
+++ "b/src/CMakeLists.txt"
@@ -56,13 +56,14 @@ if(ENABLE_CPP)
 endif(ENABLE_CPP)
 
 # Static library just for utility programs.
-SET (GD_PROGRAMS_LIB_SRC_FILES)
-
-if(MSVC)
-	set(GD_PROGRAMS_LIB_SRC_FILES ${GD_PROGRAMS_LIB_SRC_FILES}
-		getopt.c
-	)
-endif(MSVC)
+if (BUILD_PROGRAMS)
+	SET (GD_PROGRAMS_LIB_SRC_FILES)
+	if(MSVC)
+		set(GD_PROGRAMS_LIB_SRC_FILES ${GD_PROGRAMS_LIB_SRC_FILES}
+			getopt.c
+		)
+	endif(MSVC)
+endif(BUILD_PROGRAMS)
 
 include(GNUInstallDirs)
 
@@ -83,9 +84,11 @@ if (BUILD_STATIC_LIBS)
 	endif()
 endif()
 
-if (NOT "${GD_PROGRAMS_LIB_SRC_FILES}" STREQUAL "")
-	add_library(gd_programs_lib STATIC ${GD_PROGRAMS_LIB_SRC_FILES})
-endif()
+if (BUILD_PROGRAMS)
+	if (NOT "${GD_PROGRAMS_LIB_SRC_FILES}" STREQUAL "")
+		add_library(gd_programs_lib STATIC ${GD_PROGRAMS_LIB_SRC_FILES})
+	endif()
+endif(BUILD_PROGRAMS)
 
 if (WIN32 AND NOT MINGW AND NOT MSYS)
   #	SET_TARGET_PROPERTIES(${GD_LIB} PROPERTIES LINK_FLAGS "/NODEFAULTLIB:msvcrt.lib")
@@ -160,20 +163,23 @@ if (ZLIB_FOUND)
 	set(GD_PROGRAMS ${GD_PROGRAMS} gd2copypal gd2togif giftogd2)
 endif()
 
-foreach(program ${GD_PROGRAMS})
-    add_executable(${program} ${program}.c)
-
-    if (BUILD_SHARED_LIBS)
-        target_link_libraries(${program} ${GD_LIB})
-    else()
-        target_link_libraries(${program} ${GD_LIB_STATIC})
-    endif()
-	if (NOT "${GD_PROGRAMS_LIB_SRC_FILES}" STREQUAL "")
-		target_link_libraries(${program} gd_programs_lib)
-	endif()
-endforeach(program)
-
-set(GD_INSTALL_TARGETS ${GD_PROGRAMS})
+if (BUILD_PROGRAMS)
+	foreach(program ${GD_PROGRAMS})
+	    add_executable(${program} ${program}.c)
+
+	    if (BUILD_SHARED_LIBS)
+		target_link_libraries(${program} ${GD_LIB})
+	    else()
+		target_link_libraries(${program} ${GD_LIB_STATIC})
+	    endif()
+		if (NOT "${GD_PROGRAMS_LIB_SRC_FILES}" STREQUAL "")
+			target_link_libraries(${program} gd_programs_lib)
+		endif()
+	endforeach(program)
+
+	set(GD_INSTALL_TARGETS ${GD_PROGRAMS})
+	install(PROGRAMS bdftogd DESTINATION bin)
+endif()
 if (BUILD_SHARED_LIBS)
 	set(GD_INSTALL_TARGETS ${GD_INSTALL_TARGETS} ${GD_LIB})
 endif()
@@ -185,7 +191,6 @@ install(TARGETS ${GD_INSTALL_TARGETS}
         RUNTIME DESTINATION bin
         LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
         ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
-install(PROGRAMS bdftogd DESTINATION bin)
 install(FILES
 	gd.h
 	gd_color_map.h

diff --git a/src/gdhelpers.h b/src/gdhelpers.h
index 9b187af7..a6a2ad54 100644
--- a/src/gdhelpers.h
+++ b/src/gdhelpers.h
@@ -5,12 +5,16 @@
 extern "C" {
 #endif
 
-	/* sys/types.h is needed for size_t on Sparc-SunOS-4.1 */
-#ifndef _WIN32_WCE
-#include <sys/types.h>
+#ifdef _MSC_VER
+  #include <stdint.h>
 #else
-#include <stdlib.h>
-#endif /* _WIN32_WCE */
+  /* sys/types.h is needed for size_t on Sparc-SunOS-4.1 */
+  #ifndef _WIN32_WCE
+    #include <sys/types.h>
+  #else
+    #include <stdlib.h>
+  #endif /* _WIN32_WCE */
+#endif /* _WIN32 */
 
 	/* TBB: strtok_r is not universal; provide an implementation of it. */
 
