diff --git a/mod_http2/h2_mplx.c b/mod_http2/h2_mplx.c
index 8053f60..9c310c1 100644
--- a/mod_http2/h2_mplx.c
+++ b/mod_http2/h2_mplx.c
@@ -1252,10 +1252,17 @@ static apr_status_t mplx_pollset_poll(h2_mplx *m, apr_interval_time_t timeout,
             H2_MPLX_LEAVE(m);
             rv = apr_pollset_poll(m->pollset, timeout >= 0? timeout : -1, &nresults, &results);
             H2_MPLX_ENTER_ALWAYS(m);
+			// ap_log_cerror(APLOG_MARK, APLOG_ERR, rv, m->c1, APLOGNO(10310) H2_MPLX_MSG(m, "loop rv=%d eintr=%d shutdown=%d aborted=%d pcount=%d"),(int)rv,APR_STATUS_IS_EINTR(rv),m->shutdown,m->aborted,m->processing_count);
             if (APR_STATUS_IS_EINTR(rv) && m->shutdown) {
+#ifdef _WIN32
+            /* See https://github.com/icing/mod_h2/issues/277
+             * WSAPoll seems to always deliver EINTR during server graceful
+             * restarts. Treat this is an error condition */
+#else
                 if (!m->aborted) {
                     rv = APR_SUCCESS;
                 }
+#endif
                 goto cleanup;
             }
         } while (APR_STATUS_IS_EINTR(rv));
