diff --git a/src/mod_md.c b/src/mod_md.c
index 349d187..1c0c421 100644
--- a/src/mod_md.c
+++ b/src/mod_md.c
@@ -885,6 +885,35 @@ static apr_status_t md_post_config_before_ssl(apr_pool_t *p, apr_pool_t *plog,
     md_config_post_config(s, p);
     sc = md_config_get(s);
     mc = sc->mc;
+    
+    if (data != NULL) {
+        const char * const *argv;
+        const char *cmdline;
+        int exit_code;
+        if (mc->message_cmd) {
+            cmdline = apr_psprintf(p, "%s initialized MDMessageCmd", mc->message_cmd);
+            apr_tokenize_to_argv(cmdline, (char***)&argv, p);
+            rv = md_util_exec(p, argv[0], argv, &exit_code);
+        
+            if (APR_SUCCESS == rv && exit_code) rv = APR_EGENERAL;
+            if (APR_SUCCESS != rv) {
+                ap_log_error( APLOG_MARK, APLOG_INFO, 0, s, APLOGNO(10108)
+                         "MDMessageCmd %s failed with exit code %d.", mc->message_cmd, exit_code);
+            }
+        }
+        if (mc->notify_cmd) {
+            cmdline = apr_psprintf(p, "%s initialized MDNotifyCmd", mc->notify_cmd);
+            apr_tokenize_to_argv(cmdline, (char***)&argv, p);
+            rv = md_util_exec(p, argv[0], argv, &exit_code);
+        
+            if (APR_SUCCESS == rv && exit_code) rv = APR_EGENERAL;
+            if (APR_SUCCESS != rv) {
+                ap_log_error( APLOG_MARK, APLOG_INFO, 0, s, APLOGNO(10108)
+                         "MDNotifyCmd %s failed with exit code %d.", mc->notify_cmd, exit_code);
+            }
+        }
+    }
+    
     mc->dry_run = dry_run;
 
     md_event_init(p);
