diff --git a/build/generator/gen_win_dependencies.py b/build/generator/gen_win_dependencies.py
index bafc78887a..e8d71353ca 100644
--- a/build/generator/gen_win_dependencies.py
+++ b/build/generator/gen_win_dependencies.py
@@ -1519,7 +1519,7 @@ class GenDependenciesBase(gen_base.GeneratorBase):
     dll_name = None
     defines = []
 
-    lib_name = 'sqlite3.lib'
+    lib_name = 'libsqlite3.lib'
 
     if os.path.isfile(os.path.join(sqlite_base, 'include/sqlite3.h')):
       # Standard layout
