diff --git a/src/editline/readline.h b/src/editline/readline.h
index 3be13e7..fda455b 100755
--- a/src/editline/readline.h
+++ b/src/editline/readline.h
@@ -39,6 +39,14 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #ifndef _READLINE_H_
 #define _READLINE_H_
 
+#ifdef WIN32
+#    ifdef LIBRARY_EXPORTS
+#        define LIBRARY_API __declspec(dllexport)
+#    else
+#        define LIBRARY_API __declspec(dllimport)
+#    endif
+#endif
+
 #include <windows.h>
 
 /*
@@ -106,20 +114,20 @@ int history_truncate_file(const char *filename, int nlines);
 /*
 extern variables
 */
-extern char *rl_line_buffer;
-extern char *rl_prompt;
-extern int rl_point;
-extern int rl_end;
-extern int rl_attempted_completion_over;
-extern int rl_completion_append_character;
-extern int rl_readline_version;
-extern const char rl_basic_word_break_characters[];
-extern const char *rl_completer_word_break_characters;
-extern const char *rl_readline_name;
-extern const char *rl_library_version;
-extern rl_completion_func_t *rl_attempted_completion_function;
-extern rl_compentry_func_t *rl_completion_entry_function;
-extern rl_compentryfree_func_t *rl_user_completion_entry_free_function;
+extern LIBRARY_API char *rl_line_buffer;
+extern LIBRARY_API char *rl_prompt;
+extern LIBRARY_API int rl_point;
+extern LIBRARY_API int rl_end;
+extern LIBRARY_API int rl_attempted_completion_over;
+extern LIBRARY_API int rl_completion_append_character;
+extern LIBRARY_API int rl_readline_version;
+extern LIBRARY_API const char rl_basic_word_break_characters[];
+extern LIBRARY_API const char *rl_completer_word_break_characters;
+extern LIBRARY_API const char *rl_readline_name;
+extern LIBRARY_API const char *rl_library_version;
+extern LIBRARY_API rl_completion_func_t *rl_attempted_completion_function;
+extern LIBRARY_API rl_compentry_func_t *rl_completion_entry_function;
+extern LIBRARY_API rl_compentryfree_func_t *rl_user_completion_entry_free_function;
 
 #ifdef __cplusplus
 }
