diff --git "a/CMakeLists.txt" "b/CMakeLists.txt"
index d0cea9168c..1d2d7bf64d 100644
--- "a/CMakeLists.txt"
+++ "b/CMakeLists.txt"
@@ -328,12 +328,12 @@ target_include_directories(aom PUBLIC $<INSTALL_INTERFACE:include>)
 if(BUILD_SHARED_LIBS)
   add_library(aom_static STATIC ${target_objs_aom} $<TARGET_OBJECTS:aom_rtcd>)
   target_include_directories(aom_static PUBLIC $<INSTALL_INTERFACE:include>)
-  set_target_properties(aom_static PROPERTIES OUTPUT_NAME aom)
+  set_target_properties(aom_static PROPERTIES OUTPUT_NAME aom_static)
   if(MSVC OR (WIN32 AND NOT MINGW))
     # Fix race condition between the import library and the static library.
     # Affects MSVC in all three flavors (stock, clang-cl, LLVM -- the latter
     # sets MSVC and MINGW both to FALSE).
-    set_target_properties(aom PROPERTIES ARCHIVE_OUTPUT_NAME "aom_dll")
+    set_target_properties(aom PROPERTIES ARCHIVE_OUTPUT_NAME aom)
   endif()
 
   if(NOT MSVC)
