diff --git "a/lib/url.c" "b/lib/url.c"
index e8f2d12b8..f6bf6e784 100644
--- "a/lib/url.c"
+++ "b/lib/url.c"
@@ -612,6 +612,14 @@ CURLcode Curl_init_userdefined(struct Curl_easy *data)
     result = Curl_setstropt(&set->str[STRING_SSL_CAPATH_PROXY], CURL_CA_PATH);
     if(result)
       return result;
+#endif
+#if defined(USE_WIN32_CRYPTO)
+    /* Mandate Windows CA store to be used */
+    if(!set->ssl.primary.CAfile && !set->ssl.primary.CApath) {
+      /* User and environment did not specify any CA file or path.
+       */
+      set->ssl.native_ca_store = TRUE;
+    }
 #endif
   }
 
