diff --git a/lib/url.c b/lib/url.c
index aac9c2e84..17d10bee4 100644
--- a/lib/url.c
+++ b/lib/url.c
@@ -449,6 +449,14 @@ CURLcode Curl_init_userdefined(struct Curl_easy *data)
     if(result)
       return result;
 #endif
+#if defined(USE_WIN32_CRYPTO)
+    /* Mandate Windows CA store to be used */
+    if(!set->ssl.primary.CAfile && !set->ssl.primary.CApath) {
+      /* User and environment did not specify any CA file or path.
+       */
+      set->ssl.native_ca_store = TRUE;
+    } 
+#endif
 #endif
   }
 
