diff --git "a/lib/url.c" "b/lib/url.c"
index 1a95267ab..04f6f2531 100644
--- "a/lib/url.c"
+++ "b/lib/url.c"
@@ -452,6 +452,14 @@ CURLcode Curl_init_userdefined(struct Curl_easy *data)
     if(result)
       return result;
 #endif
+#if defined(USE_WIN32_CRYPTO)
+    /* Mandate Windows CA store to be used */
+    if(!set->ssl.primary.CAfile && !set->ssl.primary.CApath) {
+      /* User and environment did not specify any CA file or path.
+       */
+      set->ssl.native_ca_store = TRUE;
+    } 
+#endif
 #endif
   }
 
