diff --git "a/CMakeLists.txt" "b/CMakeLists.txt"
index 125b0a02b..e60127cb1 100644
--- "a/CMakeLists.txt"
+++ "b/CMakeLists.txt"
@@ -528,7 +528,7 @@ endfunction()
 option(USE_NGTCP2 "Use ngtcp2 and nghttp3 libraries for HTTP/3 support" OFF)
 if(USE_NGTCP2)
   if(USE_OPENSSL)
-    find_package(NGTCP2 REQUIRED OpenSSL)
+    find_package(NGTCP2 REQUIRED)
     CheckQuicSupportInOpenSSL()
   elseif(USE_GNUTLS)
     # TODO add GnuTLS support as vtls library.

diff --git a/lib/url.c b/lib/url.c
index c02d2c201..f5705c94b 100644
--- a/lib/url.c
+++ b/lib/url.c
@@ -586,6 +586,14 @@ CURLcode Curl_init_userdefined(struct Curl_easy *data)
     result = Curl_setstropt(&set->str[STRING_SSL_CAPATH_PROXY], CURL_CA_PATH);
     if(result)
       return result;
+#endif
+#if defined(USE_WIN32_CRYPTO)
+    /* Mandate Windows CA store to be used */
+    if(!set->ssl.primary.CAfile && !set->ssl.primary.CApath) {
+      /* User and environment did not specify any CA file or path.
+       */
+      set->ssl.native_ca_store = TRUE;
+    }
 #endif
   }
 
