diff --git "a/third_party/phnt/include/ntioapi.h" "b/third_party/phnt/include/ntioapi.h"
index 158e3b0..e2bcf7f 100644
--- "a/third_party/phnt/include/ntioapi.h"
+++ "b/third_party/phnt/include/ntioapi.h"
@@ -680,22 +680,6 @@ typedef struct _FILE_ID_EXTD_BOTH_DIR_INFORMATION
     WCHAR FileName[1];
 } FILE_ID_EXTD_BOTH_DIR_INFORMATION, *PFILE_ID_EXTD_BOTH_DIR_INFORMATION;
 
-// private
-typedef struct _FILE_STAT_INFORMATION
-{
-    LARGE_INTEGER FileId;
-    LARGE_INTEGER CreationTime;
-    LARGE_INTEGER LastAccessTime;
-    LARGE_INTEGER LastWriteTime;
-    LARGE_INTEGER ChangeTime;
-    LARGE_INTEGER AllocationSize;
-    LARGE_INTEGER EndOfFile;
-    ULONG FileAttributes;
-    ULONG ReparseTag;
-    ULONG NumberOfLinks;
-    ULONG EffectiveAccess;
-} FILE_STAT_INFORMATION, *PFILE_STAT_INFORMATION;
-
 // private
 typedef struct _FILE_MEMORY_PARTITION_INFORMATION
 {
