diff --git "a/gdal.cmake" "b/gdal.cmake"
index 00102cc70d..eee1fdc606 100644
--- "a/gdal.cmake"
+++ "b/gdal.cmake"
@@ -162,7 +162,7 @@ endfunction (gdal_add_private_link_libraries)
 add_library(${GDAL_LIB_TARGET_NAME} gcore/gdal.h)
 
 set(GDAL_LIB_OUTPUT_NAME
-    "gdal"
+    "libgdal"
     CACHE STRING "Name of the GDAL library")
 # If a shared lib renaming has been set in ConfigUser.cmake
 set_target_properties(${GDAL_LIB_TARGET_NAME} PROPERTIES OUTPUT_NAME ${GDAL_LIB_OUTPUT_NAME})
