diff --git a/cmake/helpers/CheckDependentLibraries.cmake b/cmake/helpers/CheckDependentLibraries.cmake
index be14b66185..f84ababfae 100644
--- a/cmake/helpers/CheckDependentLibraries.cmake
+++ b/cmake/helpers/CheckDependentLibraries.cmake
@@ -301,12 +301,12 @@ int main(){
     return 1;
 }")
       check_cxx_source_runs("${ICONV_HAS_EXTRA_CHARSETS_CODE}" ICONV_HAS_EXTRA_CHARSETS)
-      if (NOT ICONV_HAS_EXTRA_CHARSETS)
-          message(WARNING "ICONV is available but some character sets used by "
-                          "some drivers are not available. "
-                          "You may need to install an extra package "
-                          "(e.g. 'glibc-gconv-extra' on Fedora)")
-      endif()
+#      if (NOT ICONV_HAS_EXTRA_CHARSETS)
+#          message(WARNING "ICONV is available but some character sets used by "
+#                          "some drivers are not available. "
+#                          "You may need to install an extra package "
+#                          "(e.g. 'glibc-gconv-extra' on Fedora)")
+#      endif()
   endif()
 
   unset(ICONV_CONST_TEST_CODE)
From 57c92697d162cff79ef48b2f35419412b704801e Mon Sep 17 00:00:00 2001
From: Even Rouault <even.rouault@spatialys.com>
Date: Sun, 5 Feb 2023 20:36:40 +0100
Subject: [PATCH] gdallocationinfo: fix issue with VRTComplexSource and nodata
 (fix https://github.com/OSGeo/gdal/issues/7183)

---
 apps/gdallocationinfo.cpp | 16 +++++++++++-----
 1 file changed, 11 insertions(+), 5 deletions(-)

diff --git a/apps/gdallocationinfo.cpp b/apps/gdallocationinfo.cpp
index fc4132079a2..3df7fa195a2 100644
--- a/apps/gdallocationinfo.cpp
+++ b/apps/gdallocationinfo.cpp
@@ -434,14 +434,18 @@ MAIN_START(argc, argv)
             /*      Report the pixel value of this band. */
             /* --------------------------------------------------------------------
              */
-            double adfPixel[2];
+            double adfPixel[2] = {0, 0};
+            const bool bIsComplex = CPL_TO_BOOL(
+                GDALDataTypeIsComplex(GDALGetRasterDataType(hBand)));
 
             if (GDALRasterIO(hBand, GF_Read, iPixelToQuery, iLineToQuery, 1, 1,
-                             adfPixel, 1, 1, GDT_CFloat64, 0, 0) == CE_None)
+                             adfPixel, 1, 1,
+                             bIsComplex ? GDT_CFloat64 : GDT_Float64, 0,
+                             0) == CE_None)
             {
                 CPLString osValue;
 
-                if (GDALDataTypeIsComplex(GDALGetRasterDataType(hBand)))
+                if (bIsComplex)
                     osValue.Printf("%.15g+%.15gi", adfPixel[0], adfPixel[1]);
                 else
                     osValue.Printf("%.15g", adfPixel[0]);
@@ -482,11 +486,13 @@ MAIN_START(argc, argv)
                 if (dfOffset != 0.0 || dfScale != 1.0)
                 {
                     adfPixel[0] = adfPixel[0] * dfScale + dfOffset;
-                    adfPixel[1] = adfPixel[1] * dfScale + dfOffset;
 
-                    if (GDALDataTypeIsComplex(GDALGetRasterDataType(hBand)))
+                    if (bIsComplex)
+                    {
+                        adfPixel[1] = adfPixel[1] * dfScale + dfOffset;
                         osValue.Printf("%.15g+%.15gi", adfPixel[0],
                                        adfPixel[1]);
+                    }
                     else
                         osValue.Printf("%.15g", adfPixel[0]);
 
