 cmake/helpers/CheckDependentLibraries.cmake | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/cmake/helpers/CheckDependentLibraries.cmake b/cmake/helpers/CheckDependentLibraries.cmake
index 830ecef9a4..58bbdd5f75 100644
--- a/cmake/helpers/CheckDependentLibraries.cmake
+++ b/cmake/helpers/CheckDependentLibraries.cmake
@@ -68,12 +68,12 @@ int main(){
     return 1;
 }")
       check_cxx_source_runs("${ICONV_HAS_EXTRA_CHARSETS_CODE}" ICONV_HAS_EXTRA_CHARSETS)
-      if (NOT ICONV_HAS_EXTRA_CHARSETS)
-          message(WARNING "ICONV is available but some character sets used by "
-                          "some drivers are not available. "
-                          "You may need to install an extra package "
-                          "(e.g. 'glibc-gconv-extra' on Fedora)")
-      endif()
+#      if (NOT ICONV_HAS_EXTRA_CHARSETS)
+#          message(WARNING "ICONV is available but some character sets used by "
+#                          "some drivers are not available. "
+#                          "You may need to install an extra package "
+#                          "(e.g. 'glibc-gconv-extra' on Fedora)")
+#      endif()
   endif()
 
   unset(ICONV_CONST_TEST_CODE)
