diff --git a/CMakeLists.txt b/CMakeLists.txt
index 897e34c3..761625e7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -317,7 +317,9 @@ add_subdirectory(util)
 #-----------------------------------------------------------------------------
 # Documentation/Examples
 #-----------------------------------------------------------------------------
-add_subdirectory(doc)
+if(BUILD_DOC)
+  add_subdirectory(doc)
+endif()
 
 #-----------------------------------------------------------------------------
 # Install and export targets - support 'make install' or equivalent
