diff --git a/CMakeLists.txt b/CMakeLists.txt
index b78ca54b..b5398d79 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -259,12 +259,16 @@ endif()
 #-----------------------------------------------------------------------------
 # Benchmarks
 #-----------------------------------------------------------------------------
-add_subdirectory(benchmarks)
+if(BUILD_BENCHMARKS)
+  add_subdirectory(benchmarks)
+endif()
 
 #-----------------------------------------------------------------------------
 # Documentation/Examples
 #-----------------------------------------------------------------------------
-add_subdirectory(doc)
+if(BUILD_DOC)
+  add_subdirectory(doc)
+endif()
 
 #-----------------------------------------------------------------------------
 # Install and export targets - support 'make install' or equivalent
