diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7b30547d39..0a6f0e8b58 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,15 +15,28 @@
 #
 # Read README.cmake before using this.
 
+CMAKE_MINIMUM_REQUIRED(VERSION 3.5)
+
 PROJECT(HTTPD C)
 
-CMAKE_MINIMUM_REQUIRED(VERSION 2.8)
+# SET(mod_md_src "modules/md")
+SET(mod_md_src "../mod_md/src")
+# SET(mod_http2_src "modules/http2")
+SET(mod_http2_src "../mod_h2/mod_http2")
+SET(mod_suffix ".so")
+
+# https://discourse.cmake.org/t/how-to-distinguish-windows-architecture/5104
+if (CMAKE_SIZEOF_VOID_P EQUAL 4)
+    SET(diag_bits 32)
+ELSEIF(CMAKE_SIZEOF_VOID_P EQUAL 8)
+    SET(diag_bits 64)
+endif()
 
 INCLUDE(CheckSymbolExists)
 INCLUDE(CheckCSourceCompiles)
 
 FIND_PACKAGE(LibXml2)
-FIND_PACKAGE(Lua51)
+FIND_PACKAGE(Lua)
 FIND_PACKAGE(OpenSSL)
 FIND_PACKAGE(ZLIB)
 FIND_PACKAGE(CURL)
@@ -132,15 +145,15 @@ MACRO(DEFINE_WITH_BLANKS output_definition input_symbol input_value)
 ENDMACRO()
 
 MACRO(GET_MOD_ENABLE_RANK macro_modname macro_mod_enable_val macro_output_rank)
-  IF(${macro_mod_enable_val} STREQUAL "O")
+  IF(${macro_mod_enable_val} STREQUAL O)
     SET(${macro_output_rank} 0)
-  ELSEIF(${macro_mod_enable_val} STREQUAL "i")
+  ELSEIF(${macro_mod_enable_val} STREQUAL i)
     SET(${macro_output_rank} 1)
-  ELSEIF(${macro_mod_enable_val} STREQUAL "I")
+  ELSEIF(${macro_mod_enable_val} STREQUAL I)
     SET(${macro_output_rank} 2)
-  ELSEIF(${macro_mod_enable_val} STREQUAL "a")
+  ELSEIF(${macro_mod_enable_val} STREQUAL a)
     SET(${macro_output_rank} 3)
-  ELSEIF(${macro_mod_enable_val} STREQUAL "A")
+  ELSEIF(${macro_mod_enable_val} STREQUAL A)
     SET(${macro_output_rank} 4)
   ELSE()
     MESSAGE(FATAL_ERROR "Unexpected enablement value \"${macro_mod_enable_val}\" for ${macro_modname}")
@@ -242,7 +255,7 @@ MESSAGE(STATUS "")
 MESSAGE(STATUS "Summary of feature detection:")
 MESSAGE(STATUS "")
 MESSAGE(STATUS "LIBXML2_FOUND ............ : ${LIBXML2_FOUND}")
-MESSAGE(STATUS "LUA51_FOUND .............. : ${LUA51_FOUND}")
+MESSAGE(STATUS "LUA_FOUND ................ : ${LUA_FOUND}")
 MESSAGE(STATUS "NGHTTP2_FOUND ............ : ${NGHTTP2_FOUND}")
 MESSAGE(STATUS "OPENSSL_FOUND ............ : ${OPENSSL_FOUND}")
 MESSAGE(STATUS "ZLIB_FOUND ............... : ${ZLIB_FOUND}")
@@ -302,7 +315,7 @@ SET(MODULE_LIST
   "modules/cache/mod_cache_socache+I+shared object caching module"
   "modules/cache/mod_file_cache+I+File cache"
   "modules/cache/mod_socache_dbm+I+dbm small object cache provider"
-  "modules/cache/mod_socache_dc+O+distcache small object cache provider"
+# "modules/cache/mod_socache_dc+O+distcache small object cache provider"
   "modules/cache/mod_socache_memcache+I+memcache small object cache provider"
   "modules/cache/mod_socache_shmcb+I+ shmcb small object cache provider"
   "modules/cache/mod_socache_redis+I+redis small object cache provider"
@@ -343,14 +356,14 @@ SET(MODULE_LIST
   "modules/generators/mod_info+I+server information"
   "modules/generators/mod_status+I+process/thread monitoring"
   "modules/http/mod_mime+A+mapping of file-extension to MIME.  Disabling this module is normally not recommended."
-  "modules/http2/mod_http2+i+HTTP/2 protocol support"
+  "${mod_http2_src}/mod_http2+i+HTTP/2 protocol support"
   "modules/ldap/mod_ldap+i+LDAP caching and connection pooling services"
   "modules/loggers/mod_log_config+A+logging configuration.  You won't be able to log requests to the server without this module."
   "modules/loggers/mod_log_debug+I+configurable debug logging"
   "modules/loggers/mod_log_forensic+I+forensic logging"
   "modules/loggers/mod_logio+I+input and output logging"
   "modules/lua/mod_lua+i+Apache Lua Framework"
-  "modules/md/mod_md+i+Apache Managed Domains (Certificates)"
+  "${mod_md_src}/mod_md+i+Apache Managed Domains (Certificates)"
   "modules/mappers/mod_actions+I+Action triggering on requests"
   "modules/mappers/mod_alias+A+mapping of requests to different filesystem parts"
   "modules/mappers/mod_dir+A+directory request handling"
@@ -386,7 +399,7 @@ SET(MODULE_LIST
   "modules/proxy/mod_proxy_hcheck+I+Apache proxy health check module.  Requires and is enabled by --enable-proxy."
   "modules/proxy/mod_proxy_scgi+I+Apache proxy SCGI module.  Requires and is enabled by --enable-proxy."
   "modules/proxy/mod_proxy_wstunnel+I+Apache proxy Websocket Tunnel module.  Requires and is enabled by --enable-proxy."
-  "modules/http2/mod_proxy_http2+i+Apache proxy HTTP/2 module.  Requires --enable-proxy."
+  "${mod_http2_src}/mod_proxy_http2+i+Apache proxy HTTP/2 module.  Requires --enable-proxy."
   "modules/session/mod_session+I+session module"
   "modules/session/mod_session_cookie+I+session cookie module"
   "modules/session/mod_session_crypto+i+session crypto module"
@@ -399,6 +412,14 @@ SET(MODULE_LIST
   "modules/test/mod_optional_fn_import+O+example optional function importer"
   "modules/test/mod_optional_hook_export+O+example optional hook exporter"
   "modules/test/mod_optional_hook_import+O+example optional hook importer"
+  "../mod_maxminddb/src/mod_maxminddb+O+MaxMind DB Apache Module"
+  "../mod_fcgid/modules/fcgid/mod_fcgid+O+FastCGI implementation for Apache HTTP"
+  "../mod_h264_streaming/mod_h264_streaming+O+H264 Streaming Module"
+  "../mod_wku_bt/mod_whatkilledus+O+mod_whatkilledus"
+  "../mod_wku_bt/mod_crash+O+FastCGI mod_crash"
+  "../mod_wku_bt/mod_backtrace+O+mod_backtrace"
+  "../mod_bikeshed/mod_bikeshed+O+mod_bikeshed"
+  "../mod_qos/mod_qos+O+mod_qos"
 )
 
 # Track which modules actually built have APIs to link against.
@@ -453,16 +474,25 @@ SET(mod_http2_extra_defines          ssize_t=long)
 SET(mod_http2_extra_includes         ${NGHTTP2_INCLUDE_DIR})
 SET(mod_http2_extra_libs             ${NGHTTP2_LIBRARIES})
 SET(mod_http2_extra_sources
-  modules/http2/h2_bucket_beam.c     modules/http2/h2_bucket_eos.c
-  modules/http2/h2_c1.c              modules/http2/h2_c1_io.c
-  modules/http2/h2_c2.c              modules/http2/h2_c2_filter.c
-  modules/http2/h2_config.c          modules/http2/h2_conn_ctx.c
-  modules/http2/h2_mplx.c            modules/http2/h2_headers.c
-  modules/http2/h2_protocol.c        modules/http2/h2_push.c
-  modules/http2/h2_request.c         modules/http2/h2_session.c
-  modules/http2/h2_stream.c          modules/http2/h2_switch.c
-  modules/http2/h2_util.c            modules/http2/h2_workers.c
-  modules/http2/h2_ws.c
+  ${mod_http2_src}/h2_bucket_beam.c
+  ${mod_http2_src}/h2_bucket_eos.c
+  ${mod_http2_src}/h2_c1.c
+  ${mod_http2_src}/h2_c1_io.c
+  ${mod_http2_src}/h2_c2.c
+  ${mod_http2_src}/h2_c2_filter.c
+  ${mod_http2_src}/h2_config.c
+  ${mod_http2_src}/h2_conn_ctx.c
+  ${mod_http2_src}/h2_headers.c
+  ${mod_http2_src}/h2_mplx.c
+  ${mod_http2_src}/h2_protocol.c
+  ${mod_http2_src}/h2_push.c
+  ${mod_http2_src}/h2_request.c
+  ${mod_http2_src}/h2_session.c
+  ${mod_http2_src}/h2_stream.c
+  ${mod_http2_src}/h2_switch.c
+  ${mod_http2_src}/h2_util.c
+  ${mod_http2_src}/h2_workers.c
+  ${mod_http2_src}/h2_ws.c
 )
 SET(mod_ldap_extra_defines           LDAP_DECLARE_EXPORT)
 SET(mod_ldap_extra_libs              wldap32)
@@ -479,25 +509,43 @@ SET(mod_lua_extra_sources
   modules/lua/lua_passwd.c           modules/lua/lua_request.c
   modules/lua/lua_vmprep.c           modules/lua/lua_dbd.c
 )
-SET(mod_lua_requires                 LUA51_FOUND)
+SET(mod_lua_requires                 LUA_FOUND)
 SET(mod_md_requires                  OPENSSL_FOUND CURL_FOUND JANSSON_FOUND)
 SET(mod_md_extra_includes            ${OPENSSL_INCLUDE_DIR} ${CURL_INCLUDE_DIR} ${JANSSON_INCLUDE_DIR})
 SET(mod_md_extra_libs                ${OPENSSL_LIBRARIES} ${CURL_LIBRARIES} ${JANSSON_LIBRARIES} mod_watchdog)
 SET(mod_md_extra_sources
-  modules/md/md_acme.c               modules/md/md_acme_acct.c
-  modules/md/md_acme_authz.c         modules/md/md_acme_drive.c
-  modules/md/md_acmev2_drive.c       modules/md/md_event.c
-  modules/md/md_acme_order.c         modules/md/md_core.c
-  modules/md/md_curl.c               modules/md/md_crypt.c
-  modules/md/md_http.c               modules/md/md_json.c
-  modules/md/md_jws.c                modules/md/md_log.c
-  modules/md/md_result.c             modules/md/md_reg.c
-  modules/md/md_status.c             modules/md/md_store.c
-  modules/md/md_store_fs.c           modules/md/md_time.c
-  modules/md/md_ocsp.c               modules/md/md_util.c               
-  modules/md/mod_md_config.c         modules/md/mod_md_drive.c
-  modules/md/mod_md_os.c             modules/md/mod_md_status.c
-  modules/md/mod_md_ocsp.c           modules/md/md_tailscale.c
+  ${mod_md_src}/md_acme.c
+  ${mod_md_src}/md_acme_acct.c
+  ${mod_md_src}/md_acme_authz.c
+  ${mod_md_src}/md_acme_drive.c
+  ${mod_md_src}/md_acme_order.c
+  ${mod_md_src}/md_acmev2_drive.c
+  ${mod_md_src}/md_cmd_acme.c
+  ${mod_md_src}/md_cmd_main.c
+  ${mod_md_src}/md_cmd_reg.c
+  ${mod_md_src}/md_cmd_store.c
+  ${mod_md_src}/md_core.c
+  ${mod_md_src}/md_crypt.c
+  ${mod_md_src}/md_curl.c
+  ${mod_md_src}/md_event.c
+  ${mod_md_src}/md_http.c
+  ${mod_md_src}/md_json.c
+  ${mod_md_src}/md_jws.c
+  ${mod_md_src}/md_log.c
+  ${mod_md_src}/md_ocsp.c
+  ${mod_md_src}/md_reg.c
+  ${mod_md_src}/md_result.c
+  ${mod_md_src}/md_status.c
+  ${mod_md_src}/md_store.c
+  ${mod_md_src}/md_store_fs.c
+  ${mod_md_src}/md_tailscale.c
+  ${mod_md_src}/md_time.c
+  ${mod_md_src}/md_util.c
+  ${mod_md_src}/mod_md_config.c
+  ${mod_md_src}/mod_md_drive.c
+  ${mod_md_src}/mod_md_ocsp.c
+  ${mod_md_src}/mod_md_os.c
+  ${mod_md_src}/mod_md_status.c
 )
 SET(mod_optional_hook_export_extra_defines AP_DECLARE_EXPORT) # bogus reuse of core API prefix
 SET(mod_proxy_extra_defines          PROXY_DECLARE_EXPORT)
@@ -566,6 +614,50 @@ IF(LIBXML2_FOUND)
   SET(mod_xml2enc_extra_libs         "${LIBXML2_LIBRARIES};${LIBXML2_ICONV_LIBRARIES}")
 ENDIF()
 SET(mod_watchdog_extra_defines       AP_WD_DECLARE_EXPORT)
+SET(mod_fcgid_extra_sources
+../mod_fcgid/modules/fcgid/fcgid_bridge.c
+../mod_fcgid/modules/fcgid/fcgid_bucket.c
+../mod_fcgid/modules/fcgid/fcgid_conf.c
+../mod_fcgid/modules/fcgid/fcgid_filter.c
+../mod_fcgid/modules/fcgid/fcgid_mutex_unix.c
+../mod_fcgid/modules/fcgid/fcgid_pm_main.c
+#../mod_fcgid/modules/fcgid/fcgid_pm_unix.c
+../mod_fcgid/modules/fcgid/fcgid_pm_win.c
+#../mod_fcgid/modules/fcgid/fcgid_proc_unix.c
+../mod_fcgid/modules/fcgid/fcgid_proc_win.c
+#../mod_fcgid/modules/fcgid/fcgid_proctbl_unix.c
+../mod_fcgid/modules/fcgid/fcgid_proctbl_win.c
+../mod_fcgid/modules/fcgid/fcgid_protocol.c
+../mod_fcgid/modules/fcgid/fcgid_spawn_ctl.c
+)
+SET(mod_h264_streaming_extra_defines BUILDING_H264_STREAMING)
+SET(mod_h264_streaming_extra_sources
+../mod_h264_streaming/mp4_io.c
+../mod_h264_streaming/output_mp4.c
+../mod_h264_streaming/output_bucket.c
+../mod_h264_streaming/mp4_writer.c
+../mod_h264_streaming/mp4_reader.c
+../mod_h264_streaming/mp4_process.c
+../mod_h264_streaming/moov.c
+#../mod_h264_streaming/main.c
+)
+if(MAXMIND_LIBRARIES)
+  SET(mod_maxminddb_extra_libs       ${MAXMIND_LIBRARIES})
+ENDIF()
+SET(mod_whatkilledus_extra_defines   DIAG_BITS_${diag_bits})
+SET(mod_whatkilledus_extra_libs      "dbghelp.lib")
+SET(mod_whatkilledus_extra_sources
+../mod_wku_bt/diag.c          
+)
+SET(mod_backtrace_extra_defines      DIAG_BITS_${diag_bits})
+SET(mod_backtrace_extra_libs         "dbghelp.lib")
+SET(mod_backtrace_extra_sources
+../mod_wku_bt/diag.c          
+)
+SET(mod_qos_requires                 OPENSSL_FOUND)
+SET(mod_qos_extra_defines            QS_NO_STATUS_HOOK)
+SET(mod_qos_extra_includes           ${OPENSSL_INCLUDE_DIR})
+SET(mod_qos_extra_libs               ${OPENSSL_LIBRARIES})
 
 SET(MODULE_PATHS)
 FOREACH (modinfo ${MODULE_LIST})
@@ -625,14 +717,14 @@ ELSE()
 ENDIF()
 
 # for easy reference from .dll/.so builds
-CONFIGURE_FILE(os/win32/BaseAddr.ref ${PROJECT_BINARY_DIR}/ COPYONLY)
+# CONFIGURE_FILE(os/win32/BaseAddr.ref ${PROJECT_BINARY_DIR}/ COPYONLY)
 
+# https://lists.apache.org/thread/w4k1gmcynjw3l5z88mq1mfgd7g097s0f
 ADD_EXECUTABLE(gen_test_char server/gen_test_char.c)
-GET_TARGET_PROPERTY(GEN_TEST_CHAR_EXE gen_test_char LOCATION)
 ADD_CUSTOM_COMMAND(
   COMMENT "Generating character tables, test_char.h, for current locale"
   DEPENDS gen_test_char
-  COMMAND ${GEN_TEST_CHAR_EXE} > ${PROJECT_BINARY_DIR}/test_char.h
+  COMMAND "$<TARGET_FILE:gen_test_char>" > ${PROJECT_BINARY_DIR}/test_char.h
   OUTPUT ${PROJECT_BINARY_DIR}/test_char.h
 )
 ADD_CUSTOM_TARGET(
@@ -711,8 +803,8 @@ SET(HTTPD_INCLUDE_DIRECTORIES
   ${CMAKE_CURRENT_SOURCE_DIR}/modules/dav/main
   ${CMAKE_CURRENT_SOURCE_DIR}/modules/filters
   ${CMAKE_CURRENT_SOURCE_DIR}/modules/generators
-  ${CMAKE_CURRENT_SOURCE_DIR}/modules/http2
-  ${CMAKE_CURRENT_SOURCE_DIR}/modules/md
+  ${CMAKE_CURRENT_SOURCE_DIR}/${mod_http2_src}
+  ${CMAKE_CURRENT_SOURCE_DIR}/${mod_md_src}
   ${CMAKE_CURRENT_SOURCE_DIR}/modules/proxy
   ${CMAKE_CURRENT_SOURCE_DIR}/modules/session
   ${CMAKE_CURRENT_SOURCE_DIR}/modules/ssl
@@ -720,6 +812,11 @@ SET(HTTPD_INCLUDE_DIRECTORIES
   ${CMAKE_CURRENT_SOURCE_DIR}/server/mpm/winnt
   ${APR_INCLUDE_DIR}
   ${PCRE_INCLUDE_DIR}
+  ${CMAKE_CURRENT_SOURCE_DIR}/../mod_fcgid/modules/fcgid
+  ${CMAKE_CURRENT_SOURCE_DIR}/../mod_h264_streaming
+  ${CMAKE_CURRENT_SOURCE_DIR}/../mod_wku_bt
+  ${CMAKE_CURRENT_SOURCE_DIR}/../mod_bikeshed
+  ${CMAKE_CURRENT_SOURCE_DIR}/../mod_qos
 )
 
 # The .h files we install from outside the main include directory
@@ -737,7 +834,7 @@ SET(other_installed_h
   ${CMAKE_CURRENT_SOURCE_DIR}/modules/filters/mod_xml2enc.h
   ${CMAKE_CURRENT_SOURCE_DIR}/modules/generators/mod_cgi.h
   ${CMAKE_CURRENT_SOURCE_DIR}/modules/generators/mod_status.h
-  ${CMAKE_CURRENT_SOURCE_DIR}/modules/http2/mod_http2.h
+  ${CMAKE_CURRENT_SOURCE_DIR}/${mod_http2_src}/mod_http2.h
   ${CMAKE_CURRENT_SOURCE_DIR}/modules/loggers/mod_log_config.h
   ${CMAKE_CURRENT_SOURCE_DIR}/modules/mappers/mod_rewrite.h
   ${CMAKE_CURRENT_SOURCE_DIR}/modules/proxy/mod_proxy.h
@@ -802,10 +899,10 @@ FOREACH (mod ${MODULE_PATHS})
   ELSE()
     # Handle whether or not the LoadModule is commented out.
     IF(${enable_mod_val} STREQUAL "A")
-      SET(LoadModules "${LoadModules}LoadModule ${mod_module_name} modules/${mod_name}.so\n")
+      SET(LoadModules "${LoadModules}LoadModule ${mod_module_name} modules/${mod_name}${mod_suffix}\n")
       SET(mods_built_and_loaded ${mods_built_and_loaded} ${mod_name})
     ELSEIF(${enable_mod_val} STREQUAL "I")
-      SET(LoadModules "${LoadModules}# LoadModule ${mod_module_name} modules/${mod_name}.so\n")
+      SET(LoadModules "${LoadModules}# LoadModule ${mod_module_name} modules/${mod_name}${mod_suffix}\n")
       SET(mods_built_but_not_loaded ${mods_built_but_not_loaded} ${mod_name})
     ELSE()
       MESSAGE(FATAL_ERROR "${enable_mod} must be set to \"A\", \"I\", or \"O\" instead of \"${enable_mod_val}\"")
@@ -833,12 +930,12 @@ FOREACH (mod ${MODULE_PATHS})
     ENDIF()
     SET(mod_extra_libs "${mod_name}_extra_libs")
     SET_TARGET_PROPERTIES(${mod_name} PROPERTIES
-      SUFFIX .so
-      LINK_FLAGS /base:@${PROJECT_BINARY_DIR}/BaseAddr.ref,${mod_name}.so
+      SUFFIX ${mod_suffix}
+      LINK_FLAGS /dynamicbase 
     )
     TARGET_LINK_LIBRARIES(${mod_name} ${${mod_extra_libs}} libhttpd ${EXTRA_LIBS} ${APR_LIBRARIES} ${HTTPD_SYSTEM_LIBS})
     DEFINE_WITH_BLANKS(define_long_name "LONG_NAME" "${mod_name} for Apache HTTP Server")
-    SET_TARGET_PROPERTIES(${mod_name} PROPERTIES COMPILE_FLAGS "${define_long_name} -DBIN_NAME=${mod_name}.so ${EXTRA_COMPILE_FLAGS}")
+    SET_TARGET_PROPERTIES(${mod_name} PROPERTIES COMPILE_FLAGS "${define_long_name} -DBIN_NAME=${mod_name}${mod_suffix} ${EXTRA_COMPILE_FLAGS}")
 
     # Extra defines?
     SET(mod_extra_defines "${mod_name}_extra_defines")
@@ -860,7 +957,7 @@ ENDFOREACH()
 ###########   HTTPD LIBRARIES   ############
 ADD_LIBRARY(libhttpd SHARED ${LIBHTTPD_SOURCES} build/win32/httpd.rc)
 SET_TARGET_PROPERTIES(libhttpd PROPERTIES
-  LINK_FLAGS /base:@${PROJECT_BINARY_DIR}/BaseAddr.ref,libhttpd.dll
+  LINK_FLAGS /dynamicbase 
 )
 SET(install_targets ${install_targets} libhttpd)
 SET(install_bin_pdb ${install_bin_pdb} $<TARGET_PDB_FILE:libhttpd>)
@@ -915,12 +1012,12 @@ ENDIF()
 GET_PROPERTY(tmp_includes TARGET ab PROPERTY INCLUDE_DIRECTORIES)
 
 # getting duplicate manifest error with ApacheMonitor
-# ADD_EXECUTABLE(ApacheMonitor support/win32/ApacheMonitor.c support/win32/ApacheMonitor.rc)
-# SET(install_targets ${install_targets} ApacheMonitor)
-# SET(install_bin_pdb ${install_bin_pdb} $<TARGET_PDB_FILE:ApacheMonitor>)
-# SET_TARGET_PROPERTIES(ApacheMonitor PROPERTIES WIN32_EXECUTABLE TRUE)
-# SET_TARGET_PROPERTIES(ApacheMonitor PROPERTIES COMPILE_FLAGS "-DAPP_FILE -DLONG_NAME=ApacheMonitor -DBIN_NAME=ApacheMonitor.exe ${EXTRA_COMPILE_FLAGS}")
-# TARGET_LINK_LIBRARIES(ApacheMonitor ${EXTRA_LIBS} ${HTTPD_SYSTEM_LIBS} comctl32 wtsapi32)
+ADD_EXECUTABLE(ApacheMonitor support/win32/ApacheMonitor.c support/win32/ApacheMonitor.rc)
+SET(install_targets ${install_targets} ApacheMonitor)
+SET(install_bin_pdb ${install_bin_pdb} $<TARGET_PDB_FILE:ApacheMonitor>)
+SET_TARGET_PROPERTIES(ApacheMonitor PROPERTIES WIN32_EXECUTABLE TRUE)
+SET_TARGET_PROPERTIES(ApacheMonitor PROPERTIES COMPILE_FLAGS "-DAPP_FILE -DLONG_NAME=ApacheMonitor -DBIN_NAME=ApacheMonitor.exe ${EXTRA_COMPILE_FLAGS}")
+TARGET_LINK_LIBRARIES(ApacheMonitor ${EXTRA_LIBS} ${HTTPD_SYSTEM_LIBS} comctl32 wtsapi32)
 
 ###########  CONFIGURATION FILES ###########
 # Set up variables used in the .conf file templates
diff --git a/modules/proxy/mod_proxy_fcgi.c b/modules/proxy/mod_proxy_fcgi.c
index d420df6a77..aca014002a 100644
--- a/modules/proxy/mod_proxy_fcgi.c
+++ b/modules/proxy/mod_proxy_fcgi.c
@@ -381,6 +381,10 @@ static apr_status_t send_environment(proxy_conn_rec *conn, request_rec *r,
 
         if (newfname) {
             newfname = ap_strchr(newfname, '/');
+			// Windows
+			if ((ap_strstr(newfname, ":/")) != NULL)
+				memmove(newfname, newfname+1, strlen(newfname));
+			ap_log_rerror(APLOG_MARK, APLOG_DEBUG, 0, r, APLOGNO(01078) "SCRIPT_FILENAME %s", newfname);
             r->filename = newfname;
         }
     }
diff --git a/server/main.c b/server/main.c
index fb23f01d1d..7d42ec8128 100644
--- a/server/main.c
+++ b/server/main.c
@@ -45,6 +45,13 @@
 #if APR_HAVE_UNISTD_H
 #include <unistd.h>
 #endif
+/* The following logic ensures we correctly glue FILE* within one CRT used
+ * by the OpenSSL library build to another CRT used by the ab.exe build.
+ * This became especially problematic with Visual Studio 2015.
+ * https://www.apachelounge.com/viewtopic.php?p=33600 
+ * C:\sdk\src\openssl\ms\applink.c
+*/
+#include <ms/applink.c>
 
 /* WARNING: Win32 binds http_main.c dynamically to the server. Please place
  *          extern functions and global data in another appropriate module.
diff --git a/support/ab.c b/support/ab.c
index 6709cd1db6..22e0f68afd 100644
--- a/support/ab.c
+++ b/support/ab.c
@@ -189,7 +189,7 @@ typedef STACK_OF(X509) X509_STACK_TYPE;
  * by the OpenSSL library build to another CRT used by the ab.exe build.
  * This became especially problematic with Visual Studio 2015.
  */
-#include <openssl/applink.c>
+#include <ms/applink.c>
 #endif
 
 #if (OPENSSL_VERSION_NUMBER >= 0x00909000)
diff --git a/support/win32/ApacheMonitor.rc b/support/win32/ApacheMonitor.rc
index d4203bf926..053f77522e 100644
--- a/support/win32/ApacheMonitor.rc
+++ b/support/win32/ApacheMonitor.rc
@@ -18,6 +18,7 @@
 
 #include "ApacheMonitor.h"
 
+#undef  LONG_NAME
 #define LONG_NAME Apache HTTP Server Monitor
 
 #include "../../build/win32/httpd.rc"
@@ -26,7 +27,7 @@
 IDI_APSRVMON            ICON    DISCARDABLE     "ApacheMonitor.ico"
 IDI_ICOSTOP             ICON    DISCARDABLE     "apstop.ico"
 IDI_ICORUN              ICON    DISCARDABLE     "aprun.ico"
-CREATEPROCESS_MANIFEST_RESOURCE_ID RT_MANIFEST  "ApacheMonitor.manifest"
+// CREATEPROCESS_MANIFEST_RESOURCE_ID RT_MANIFEST  "ApacheMonitor.manifest"
 
 IDD_DLGSERVICES DIALOGEX 0, 0, 350, 192
 STYLE DS_MODALFRAME | DS_SETFOREGROUND | WS_MINIMIZEBOX | WS_VISIBLE | 
