diff --git a/configure.ac b/configure.ac
index 86c3a38..fdaa123 100644
--- a/configure.ac
+++ b/configure.ac
@@ -73,7 +73,7 @@ fi;
 
 
 dnl Checks for header files.
-AC_CHECK_INCLUDES_DEFAULT
+# AC_CHECK_INCLUDES_DEFAULT
 
 AC_CHECK_HEADERS(unistd.h stdint.h xlocale.h)
 
 lib/grammar.c | 74 +++++++++++++++++++++++++++++------------------------------
 lib/grammar.h | 10 ++------
 2 files changed, 39 insertions(+), 45 deletions(-)

diff --git a/lib/grammar.c b/lib/grammar.c
index 26aa039..ce187d9 100644
--- a/lib/grammar.c
+++ b/lib/grammar.c
@@ -1,4 +1,4 @@
-/* A Bison parser, made by GNU Bison 3.8.  */
+/* A Bison parser, made by GNU Bison 3.8.2.  */
 
 /* Bison implementation for Yacc-like parsers in C
 
@@ -46,10 +46,10 @@
    USER NAME SPACE" below.  */
 
 /* Identify Bison output, and Bison version.  */
-#define YYBISON 30800
+#define YYBISON 30802
 
 /* Bison version string.  */
-#define YYBISON_VERSION "3.8"
+#define YYBISON_VERSION "3.8.2"
 
 /* Skeleton name.  */
 #define YYSKELETON_NAME "yacc.c"
@@ -84,8 +84,7 @@
 #include "wincompat.h"
 
 /* These declarations are provided to suppress compiler warnings. */
-extern int libconfig_yylex();
-extern int libconfig_yyget_lineno();
+extern int libconfig_yyget_lineno(void *);
 
 #define YYMALLOC libconfig_malloc
 
@@ -117,7 +116,7 @@ void libconfig_yyerror(void *scanner, struct parse_context *ctx,
 }
 
 
-#line 121 "grammar.c"
+#line 120 "grammar.c"
 
 # ifndef YY_CAST
 #  ifdef __cplusplus
@@ -214,14 +213,14 @@ extern int libconfig_yydebug;
 #if ! defined YYSTYPE && ! defined YYSTYPE_IS_DECLARED
 union YYSTYPE
 {
-#line 78 "grammar.y"
+#line 77 "grammar.y"
 
   int ival;
   long long llval;
   double fval;
   char *sval;
 
-#line 225 "grammar.c"
+#line 224 "grammar.c"
 
 };
 typedef union YYSTYPE YYSTYPE;
@@ -231,12 +230,6 @@ typedef union YYSTYPE YYSTYPE;
 
 
 
-#if !defined libconfig_yyerror && !defined YYERROR_IS_DECLARED
-void libconfig_yyerror (void *scanner, struct parse_context *ctx, struct scan_context *scan_ctx, const char *msg);
-#endif
-#if !defined libconfig_yylex && !defined YYLEX_IS_DECLARED
-int libconfig_yylex (YYSTYPE *yylvalp, void *scanner);
-#endif
 
 int libconfig_yyparse (void *scanner, struct parse_context *ctx, struct scan_context *scan_ctx);
 
@@ -293,6 +286,13 @@ enum yysymbol_kind_t
 typedef enum yysymbol_kind_t yysymbol_kind_t;
 
 
+/* Second part of user prologue.  */
+#line 84 "grammar.y"
+
+/* These declarations are provided to suppress compiler warnings. */
+extern int libconfig_yylex(YYSTYPE *, void *);
+
+#line 296 "grammar.c"
 
 
 #ifdef short
@@ -676,11 +676,11 @@ static const yytype_int8 yytranslate[] =
 /* YYRLINE[YYN] -- Source line where rule number YYN was defined.  */
 static const yytype_int16 yyrline[] =
 {
-       0,    94,    94,    96,   100,   101,   104,   106,   109,   111,
-     112,   117,   116,   136,   135,   159,   158,   181,   182,   183,
-     184,   188,   189,   193,   213,   235,   257,   279,   301,   319,
-     347,   348,   349,   352,   354,   358,   359,   360,   363,   365,
-     370,   369
+       0,    98,    98,   100,   104,   105,   108,   110,   113,   115,
+     116,   121,   120,   140,   139,   163,   162,   185,   186,   187,
+     188,   192,   193,   197,   217,   239,   261,   283,   305,   323,
+     351,   352,   353,   356,   358,   362,   363,   364,   367,   369,
+     374,   373
 };
 #endif
 
@@ -1019,7 +1019,7 @@ yydestruct (const char *yymsg,
   switch (yykind)
     {
     case YYSYMBOL_TOK_STRING: /* TOK_STRING  */
-#line 90 "grammar.y"
+#line 94 "grammar.y"
             { free(((*yyvaluep).sval)); }
 #line 1025 "grammar.c"
         break;
@@ -1295,7 +1295,7 @@ yyreduce:
   switch (yyn)
     {
   case 11: /* $@1: %empty  */
-#line 117 "grammar.y"
+#line 121 "grammar.y"
   {
     ctx->setting = config_setting_add(ctx->parent, (yyvsp[0].sval), CONFIG_TYPE_NONE);
 
@@ -1313,7 +1313,7 @@ yyreduce:
     break;
 
   case 13: /* $@2: %empty  */
-#line 136 "grammar.y"
+#line 140 "grammar.y"
   {
     if(IN_LIST())
     {
@@ -1331,7 +1331,7 @@ yyreduce:
     break;
 
   case 14: /* array: TOK_ARRAY_START $@2 simple_value_list_optional TOK_ARRAY_END  */
-#line 151 "grammar.y"
+#line 155 "grammar.y"
   {
     if(ctx->parent)
       ctx->parent = ctx->parent->parent;
@@ -1340,7 +1340,7 @@ yyreduce:
     break;
 
   case 15: /* $@3: %empty  */
-#line 159 "grammar.y"
+#line 163 "grammar.y"
   {
     if(IN_LIST())
     {
@@ -1358,7 +1358,7 @@ yyreduce:
     break;
 
   case 16: /* list: TOK_LIST_START $@3 value_list_optional TOK_LIST_END  */
-#line 174 "grammar.y"
+#line 178 "grammar.y"
   {
     if(ctx->parent)
       ctx->parent = ctx->parent->parent;
@@ -1367,19 +1367,19 @@ yyreduce:
     break;
 
   case 21: /* string: TOK_STRING  */
-#line 188 "grammar.y"
+#line 192 "grammar.y"
              { libconfig_parsectx_append_string(ctx, (yyvsp[0].sval)); free((yyvsp[0].sval)); }
 #line 1373 "grammar.c"
     break;
 
   case 22: /* string: string TOK_STRING  */
-#line 189 "grammar.y"
+#line 193 "grammar.y"
                       { libconfig_parsectx_append_string(ctx, (yyvsp[0].sval)); free((yyvsp[0].sval)); }
 #line 1379 "grammar.c"
     break;
 
   case 23: /* simple_value: TOK_BOOLEAN  */
-#line 194 "grammar.y"
+#line 198 "grammar.y"
   {
     if(IN_ARRAY() || IN_LIST())
     {
@@ -1403,7 +1403,7 @@ yyreduce:
     break;
 
   case 24: /* simple_value: TOK_INTEGER  */
-#line 214 "grammar.y"
+#line 218 "grammar.y"
   {
     if(IN_ARRAY() || IN_LIST())
     {
@@ -1429,7 +1429,7 @@ yyreduce:
     break;
 
   case 25: /* simple_value: TOK_INTEGER64  */
-#line 236 "grammar.y"
+#line 240 "grammar.y"
   {
     if(IN_ARRAY() || IN_LIST())
     {
@@ -1455,7 +1455,7 @@ yyreduce:
     break;
 
   case 26: /* simple_value: TOK_HEX  */
-#line 258 "grammar.y"
+#line 262 "grammar.y"
   {
     if(IN_ARRAY() || IN_LIST())
     {
@@ -1481,7 +1481,7 @@ yyreduce:
     break;
 
   case 27: /* simple_value: TOK_HEX64  */
-#line 280 "grammar.y"
+#line 284 "grammar.y"
   {
     if(IN_ARRAY() || IN_LIST())
     {
@@ -1507,7 +1507,7 @@ yyreduce:
     break;
 
   case 28: /* simple_value: TOK_FLOAT  */
-#line 302 "grammar.y"
+#line 306 "grammar.y"
   {
     if(IN_ARRAY() || IN_LIST())
     {
@@ -1529,7 +1529,7 @@ yyreduce:
     break;
 
   case 29: /* simple_value: string  */
-#line 320 "grammar.y"
+#line 324 "grammar.y"
   {
     if(IN_ARRAY() || IN_LIST())
     {
@@ -1558,7 +1558,7 @@ yyreduce:
     break;
 
   case 40: /* $@4: %empty  */
-#line 370 "grammar.y"
+#line 374 "grammar.y"
   {
     if(IN_LIST())
     {
@@ -1576,7 +1576,7 @@ yyreduce:
     break;
 
   case 41: /* group: TOK_GROUP_START $@4 setting_list_optional TOK_GROUP_END  */
-#line 385 "grammar.y"
+#line 389 "grammar.y"
   {
     if(ctx->parent)
       ctx->parent = ctx->parent->parent;
@@ -1778,5 +1778,5 @@ yyreturnlab:
   return yyresult;
 }
 
-#line 391 "grammar.y"
+#line 395 "grammar.y"
 
diff --git a/lib/grammar.h b/lib/grammar.h
index 84fa1c9..6d76aaf 100644
--- a/lib/grammar.h
+++ b/lib/grammar.h
@@ -1,4 +1,4 @@
-/* A Bison parser, made by GNU Bison 3.8.  */
+/* A Bison parser, made by GNU Bison 3.8.2.  */
 
 /* Bison interface for Yacc-like parsers in C
 
@@ -107,7 +107,7 @@ extern int libconfig_yydebug;
 #if ! defined YYSTYPE && ! defined YYSTYPE_IS_DECLARED
 union YYSTYPE
 {
-#line 78 "grammar.y"
+#line 77 "grammar.y"
 
   int ival;
   long long llval;
@@ -124,12 +124,6 @@ typedef union YYSTYPE YYSTYPE;
 
 
 
-#if !defined libconfig_yyerror && !defined YYERROR_IS_DECLARED
-void libconfig_yyerror (void *scanner, struct parse_context *ctx, struct scan_context *scan_ctx, const char *msg);
-#endif
-#if !defined libconfig_yylex && !defined YYLEX_IS_DECLARED
-int libconfig_yylex (YYSTYPE *yylvalp, void *scanner);
-#endif
 
 int libconfig_yyparse (void *scanner, struct parse_context *ctx, struct scan_context *scan_ctx);
 
