 expat/lib/expat.h | 46 +++++++++++++++++++++++-----------------------
 1 file changed, 23 insertions(+), 23 deletions(-)

diff --git a/expat/lib/expat.h b/expat/lib/expat.h
index 290dfeb0..5a52fbee 100644
--- a/expat/lib/expat.h
+++ b/expat/lib/expat.h
@@ -43,21 +43,21 @@
 */
 
 #ifndef Expat_INCLUDED
-#  define Expat_INCLUDED 1
+#define Expat_INCLUDED 1
 
-#  include <stdlib.h>
-#  include "expat_external.h"
+#include <stdlib.h>
+#include "expat_external.h"
 
-#  ifdef __cplusplus
+#ifdef __cplusplus
 extern "C" {
-#  endif
+#endif
 
 struct XML_ParserStruct;
 typedef struct XML_ParserStruct *XML_Parser;
 
 typedef unsigned char XML_Bool;
-#  define XML_TRUE ((XML_Bool)1)
-#  define XML_FALSE ((XML_Bool)0)
+#define XML_TRUE ((XML_Bool)1)
+#define XML_FALSE ((XML_Bool)0)
 
 /* The XML_Status enum gives the possible return values for several
    API functions.  The preprocessor #defines are included so this
@@ -74,11 +74,11 @@ typedef unsigned char XML_Bool;
 */
 enum XML_Status {
   XML_STATUS_ERROR = 0,
-#  define XML_STATUS_ERROR XML_STATUS_ERROR
+#define XML_STATUS_ERROR XML_STATUS_ERROR
   XML_STATUS_OK = 1,
-#  define XML_STATUS_OK XML_STATUS_OK
+#define XML_STATUS_OK XML_STATUS_OK
   XML_STATUS_SUSPENDED = 2
-#  define XML_STATUS_SUSPENDED XML_STATUS_SUSPENDED
+#define XML_STATUS_SUSPENDED XML_STATUS_SUSPENDED
 };
 
 enum XML_Error {
@@ -681,7 +681,7 @@ XMLPARSEAPI(void)
 XML_SetUserData(XML_Parser parser, void *userData);
 
 /* Returns the last value set by XML_SetUserData or NULL. */
-#  define XML_GetUserData(parser) (*(void **)(parser))
+#define XML_GetUserData(parser) (*(void **)(parser))
 
 /* This is equivalent to supplying an encoding argument to
    XML_ParserCreate. On success XML_SetEncoding returns non-zero,
@@ -753,7 +753,7 @@ XML_GetSpecifiedAttributeCount(XML_Parser parser);
 XMLPARSEAPI(int)
 XML_GetIdAttributeIndex(XML_Parser parser);
 
-#  ifdef XML_ATTR_INFO
+#ifdef XML_ATTR_INFO
 /* Source file byte offsets for the start and end of attribute names and values.
    The value indices are exclusive of surrounding quotes; thus in a UTF-8 source
    file an attribute value of "blah" will yield:
@@ -774,7 +774,7 @@ typedef struct {
 */
 XMLPARSEAPI(const XML_AttrInfo *)
 XML_GetAttributeInfo(XML_Parser parser);
-#  endif
+#endif
 
 /* Parses some input. Returns XML_STATUS_ERROR if a fatal error is
    detected.  The last call to XML_Parse must have isFinal true; len
@@ -971,9 +971,9 @@ XMLPARSEAPI(const char *)
 XML_GetInputContext(XML_Parser parser, int *offset, int *size);
 
 /* For backwards compatibility with previous versions. */
-#  define XML_GetErrorLineNumber XML_GetCurrentLineNumber
-#  define XML_GetErrorColumnNumber XML_GetCurrentColumnNumber
-#  define XML_GetErrorByteIndex XML_GetCurrentByteIndex
+#define XML_GetErrorLineNumber XML_GetCurrentLineNumber
+#define XML_GetErrorColumnNumber XML_GetCurrentColumnNumber
+#define XML_GetErrorByteIndex XML_GetCurrentByteIndex
 
 /* Frees the content model passed to the element declaration handler */
 XMLPARSEAPI(void)
@@ -1049,7 +1049,7 @@ typedef struct {
 XMLPARSEAPI(const XML_Feature *)
 XML_GetFeatureList(void);
 
-#  if defined(XML_DTD) || (defined(XML_GE) && XML_GE == 1)
+#if defined(XML_DTD) || (defined(XML_GE) && XML_GE == 1)
 /* Added in Expat 2.4.0 for XML_DTD defined and
  * added in Expat 2.6.0 for XML_GE == 1. */
 XMLPARSEAPI(XML_Bool)
@@ -1071,7 +1071,7 @@ XML_SetAllocTrackerMaximumAmplification(XML_Parser parser,
 XMLPARSEAPI(XML_Bool)
 XML_SetAllocTrackerActivationThreshold(
     XML_Parser parser, unsigned long long activationThresholdBytes);
-#  endif
+#endif
 
 /* Added in Expat 2.6.0. */
 XMLPARSEAPI(XML_Bool)
@@ -1080,12 +1080,12 @@ XML_SetReparseDeferralEnabled(XML_Parser parser, XML_Bool enabled);
 /* Expat follows the semantic versioning convention.
    See https://semver.org
 */
-#  define XML_MAJOR_VERSION 2
-#  define XML_MINOR_VERSION 7
-#  define XML_MICRO_VERSION 3
+#define XML_MAJOR_VERSION 2
+#define XML_MINOR_VERSION 7
+#define XML_MICRO_VERSION 3
 
-#  ifdef __cplusplus
+#ifdef __cplusplus
 }
-#  endif
+#endif
 
 #endif /* not Expat_INCLUDED */
