diff --git a/libheif/CMakeLists.txt b/libheif/CMakeLists.txt
index d47c908..1adf84c 100644
--- a/libheif/CMakeLists.txt
+++ b/libheif/CMakeLists.txt
@@ -70,9 +70,12 @@ if (ENABLE_PLUGIN_LOADING AND UNIX)
 endif()
 
 if (WITH_DEFLATE_HEADER_COMPRESSION)
-    find_library(z-lib z)
-    target_link_libraries(heif PRIVATE ${z-lib})
-    target_compile_definitions(heif PRIVATE WITH_DEFLATE_HEADER_COMPRESSION=1)
+find_package( ZLIB REQUIRED )
+	if ( ZLIB_FOUND )
+	    target_link_libraries(heif PRIVATE ${ZLIB_LIBRARY})
+	    target_compile_definitions(heif PRIVATE WITH_DEFLATE_HEADER_COMPRESSION=1)
+	    target_include_directories(heif PRIVATE ${ZLIB_INCLUDE_DIRS})
+	endif ()
 endif ()
 
 write_basic_package_version_file(${PROJECT_NAME}-config-version.cmake COMPATIBILITY ExactVersion)
