diff --git "a/CMakeLists.txt" "b/CMakeLists.txt"
index 1cdc524..325024b 100644
--- "a/CMakeLists.txt"
+++ "b/CMakeLists.txt"
@@ -88,11 +88,11 @@ if(NOT LIBSSH2_VERSION OR
 endif()
 
 include(GNUInstallDirs)
-install(
-  FILES
-    COPYING NEWS README RELEASE-NOTES
-    docs/AUTHORS docs/BINDINGS.md docs/HACKING.md
-  DESTINATION ${CMAKE_INSTALL_DOCDIR})
+# install(
+#   FILES
+#     COPYING NEWS README RELEASE-NOTES
+#     docs/AUTHORS docs/BINDINGS.md docs/HACKING.md
+#  DESTINATION ${CMAKE_INSTALL_DOCDIR})
 
 include(max_warnings)
 
@@ -315,6 +315,7 @@ if(CRYPTO_BACKEND STREQUAL "OpenSSL" OR NOT CRYPTO_BACKEND)
         NAMES crypto.dll
           libcrypto-1_1.dll libcrypto-1_1-x64.dll
           libcrypto-3.dll libcrypto-3-x64.dll
+	  libcrypto-81_3.dll libcrypto-81_3-x64.dll
         HINTS ${_OPENSSL_ROOT_HINTS} PATHS ${_OPENSSL_ROOT_PATHS}
         PATH_SUFFIXES bin NO_DEFAULT_PATH)
       if(DLL_LIBCRYPTO)
@@ -328,6 +329,7 @@ if(CRYPTO_BACKEND STREQUAL "OpenSSL" OR NOT CRYPTO_BACKEND)
         NAMES ssl.dll
           libssl-1_1.dll libssl-1_1-x64.dll
           libssl-3.dll libssl-3-x64.dll
+	  libssl-81_3.dll libssl-81_3-x64.dll
         HINTS ${_OPENSSL_ROOT_HINTS} PATHS ${_OPENSSL_ROOT_PATHS}
         PATH_SUFFIXES bin NO_DEFAULT_PATH)
       if(DLL_LIBSSL)
@@ -459,7 +461,7 @@ if(LINT)
   endif()
 endif()
 
-add_subdirectory(docs)
+# add_subdirectory(docs)
 
 feature_summary(WHAT ALL)
 
