diff --git a/include/libyuv/planar_functions.h b/include/libyuv/planar_functions.h
index f7c6db80..2769896e 100644
--- a/include/libyuv/planar_functions.h
+++ b/include/libyuv/planar_functions.h
@@ -106,6 +106,7 @@ void DetileSplitUVPlane(const uint8_t* src_uv,
                         int tile_height);
 
 // Convert a Y and UV plane of tiles into interlaced YUY2.
+LIBYUV_API
 void DetileToYUY2(const uint8_t* src_y,
                   int src_stride_y,
                   const uint8_t* src_uv,
