diff --git "a/CMakeLists.txt" "b/CMakeLists.txt"
index 6f372949..6222240a 100644
--- "a/CMakeLists.txt"
+++ "b/CMakeLists.txt"
@@ -24,17 +24,17 @@ INCLUDE_DIRECTORIES( BEFORE ${ly_inc_dir} )
 
 if(MSVC)
   ADD_DEFINITIONS ( -D_CRT_SECURE_NO_WARNINGS )
+else()
+  target_compile_options(${ly_lib_static} PRIVATE
+    -Wno-sign-compare
+    -Wno-unused-function
+    -Wunused-variable
+    -Wuninitialized)
 endif()
 
 # this creates the static library (.a)
 ADD_LIBRARY				( ${ly_lib_static} STATIC ${ly_source_files} )
 
-target_compile_options(${ly_lib_static} PRIVATE
-  -Wno-sign-compare
-  -Wno-unused-function
-  -Wunused-variable
-  -Wuninitialized)
-
 # this creates the shared library (.so)
 ADD_LIBRARY				( ${ly_lib_shared} SHARED ${ly_source_files} )
 SET_TARGET_PROPERTIES	( ${ly_lib_shared} PROPERTIES OUTPUT_NAME "${ly_lib_name}" )
