diff --git a/mapsforgesrv/build.gradle b/mapsforgesrv/build.gradle
index 312897d..5610c69 100644
--- a/mapsforgesrv/build.gradle
+++ b/mapsforgesrv/build.gradle
@@ -28,7 +28,7 @@ dependencies {
     implementation 'com.google.guava:guava:30.1.1-jre'
 
     // Use JUnit test framework
-    testImplementation 'junit:junit:4.12'
+    testImplementation 'junit:junit:4.13.2'
     
     implementation group: 'org.apache.logging.log4j', name: 'log4j-core', version: '2.14.1'
     implementation group: 'org.apache.logging.log4j', name: 'log4j-api', version: '2.14.1'
@@ -42,7 +42,7 @@ dependencies {
     implementation 'org.mapsforge:mapsforge-poi-awt:0.16.0'
     implementation 'org.mapsforge:mapsforge-poi:0.16.0'
 
-    implementation 'org.xerial:sqlite-jdbc:3.36.0.1'
+    implementation 'org.xerial:sqlite-jdbc:3.36.0.3'
 
     implementation 'net.sf.kxml:kxml2:2.3.0' 
     implementation group: 'org.eclipse.jetty', name: 'jetty-server', version: '11.0.6'
diff --git a/mapsforgesrv/gradle/wrapper/gradle-wrapper.properties b/mapsforgesrv/gradle/wrapper/gradle-wrapper.properties
index 05679dc..ffed3a2 100644
--- a/mapsforgesrv/gradle/wrapper/gradle-wrapper.properties
+++ b/mapsforgesrv/gradle/wrapper/gradle-wrapper.properties
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.1.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.2-bin.zip
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
