diff --git a/apache2/Makefile.win b/apache2/Makefile.win
index ed4bfc9e..e961e80c 100644
--- a/apache2/Makefile.win
+++ b/apache2/Makefile.win
@@ -10,9 +10,9 @@
 LIBS = $(APACHE)\lib\libhttpd.lib \
        $(APACHE)\lib\libapr-1.lib \
        $(APACHE)\lib\libaprutil-1.lib \
-       $(PCRE)\pcre.lib \
-       $(CURL)\libcurl.lib \
-       $(LIBXML2)\win32\bin.msvc\libxml2.lib \
+       $(PCRE)\lib\pcre2-8.lib \
+       $(CURL)\lib\libcurl.lib \
+       $(LIBXML2)\lib\libxml2.lib \
        Ws2_32.lib \
        "iphlpapi.lib"
 
@@ -20,30 +20,34 @@ LIBS = $(APACHE)\lib\libhttpd.lib \
 ###########################################################################
 
 
-!IF "$(IIS_BUILD)" == "yes"
-DEFS=$(DEFS) -DVERSION_IIS
-!ENDIF
+!IF "$(IIS_BUILD)" == "yes"
+
+DEFS=$(DEFS) -DVERSION_IIS
+
+!ENDIF
+
 
 CC = CL
 
 MT = mt
 
-DEFS = /nologo /O2 /LD /W3 /wd4244 /wd4018 -DWIN32 -DWINNT -Dinline=APR_INLINE -D$(VERSION)
+DEFS = /nologo /DWIN32 /DWINNT /D$(VERSION) /D_WINDOWS /DWITH_PCRE_JIT /w /Zf /Zi /FS /O2 /GL /MD /DNDEBUG /Ob3
 
 DLL = mod_security2.so
 
 INCLUDES = -I. -I.. \
            -I$(CURL)\include -I$(CURL) \
            -I$(PCRE)\include -I$(PCRE) \
-           -I$(LIBXML2)\include \
+           -I$(LIBXML2)\include\libxml2 \
            -I$(APACHE)\include
 
-# Enables support for SecRemoteRules and external resources.
+# Enables support for SecRemoteRules and external resources.
+
 DEFS=$(DEFS) -DWITH_CURL -DWITH_REMOTE_RULES
 
 # Lua is optional
 !IF "$(LUA)" != ""
-LIBS = $(LIBS) $(LUA)\lua5.1.lib
+LIBS = $(LIBS) $(LUA)\lib\liblua.lib
 DEFS=$(DEFS) -DWITH_LUA
 INCLUDES = $(INCLUDES) -I$(LUA)\include -I$(LUA) \
 !ENDIF
diff --git a/apache2/mod_security2.c b/apache2/mod_security2.c
index 0dcb6b3e..36c896cb 100644
--- a/apache2/mod_security2.c
+++ b/apache2/mod_security2.c
@@ -117,25 +117,25 @@ static void version(apr_pool_t *mp, server_rec* s) {
     }
 
 #ifndef WITH_PCRE
-    pcre_vrs = apr_psprintf(mp,"%d.%d ", PCRE2_MAJOR, PCRE2_MINOR);
+    pcre_vrs = apr_psprintf(mp,"%d.%d", PCRE2_MAJOR, PCRE2_MINOR);
     pcre_loaded_vrs = pcre2_loaded_vrs_buffer;
     pcre2_config(PCRE2_CONFIG_VERSION, pcre2_loaded_vrs_buffer);
 #else
-    pcre_vrs = apr_psprintf(mp,"%d.%d ", PCRE_MAJOR, PCRE_MINOR);
+    pcre_vrs = apr_psprintf(mp,"%d.%d", PCRE_MAJOR, PCRE_MINOR);
     pcre_loaded_vrs = pcre_version();
 #endif
 
-    ap_log_error(APLOG_MARK, APLOG_NOTICE, 0, NULL,
 #ifndef WITH_PCRE
-            "ModSecurity: PCRE2 compiled version=\"%s\"; "
+            ap_log_error(APLOG_MARK, APLOG_NOTICE, 0, NULL,"ModSecurity: PCRE2 compiled version=\"%s\"; loaded version=\"%s\"", pcre_vrs, pcre_loaded_vrs);
 #else
-            "ModSecurity: PCRE compiled version=\"%s\"; "
+            ap_log_error(APLOG_MARK, APLOG_NOTICE, 0, NULL,"ModSecurity: PCRE compiled version=\"%s\"; loaded version=\"%s\"", pcre_vrs, pcre_loaded_vrs);
 #endif
-            "loaded version=\"%s\"", pcre_vrs, pcre_loaded_vrs);
 
-    if (strstr(pcre_loaded_vrs,pcre_vrs) == NULL)    {
+    /* failed on PCRE-dev: PCRE compiled version="10.46 "; loaded version="10.46-DEV 2024-06-09"
+	if (strstr(pcre_loaded_vrs,pcre_vrs) == NULL)    {
         ap_log_error(APLOG_MARK, APLOG_WARNING, 0, s, "ModSecurity: Loaded PCRE do not match with compiled!");
     }
+	*/
 
     /* Lua version function was removed in current 5.1. Need to check in future versions if it's back */
 #if defined(WITH_LUA)
@@ -775,7 +775,7 @@ static int hook_post_config(apr_pool_t *mp, apr_pool_t *mp_log, apr_pool_t *mp_t
 
     /* Log our presence to the error log. */
     if (first_time) {
-        ap_log_error(APLOG_MARK, APLOG_NOTICE | APLOG_NOERRNO, 0, s,
+        ap_log_error(APLOG_MARK, APLOG_INFO | APLOG_NOERRNO, 0, s,
                 "%s configured.", MODSEC_MODULE_NAME_FULL);
 
         version(mp, s);
