Index: modules/fcgid/fcgid_bridge.c
===================================================================
--- modules/fcgid/fcgid_bridge.c	(revision 1877889)
+++ modules/fcgid/fcgid_bridge.c	(working copy)
@@ -30,7 +30,7 @@
 #include "fcgid_spawn_ctl.h"
 #include "fcgid_protocol.h"
 #include "fcgid_bucket.h"
-#define FCGID_APPLY_TRY_COUNT 2
+#define FCGID_APPLY_TRY_COUNT 4
 #define FCGID_REQUEST_COUNT 32
 #define FCGID_BRIGADE_CLEAN_STEP 32
 
@@ -452,6 +452,10 @@
             /* Avoid sleeping the very first time through if there are no
                busy processes; the problem is just that we haven't spawned
                anything yet, so waiting is pointless */
+
+/*			https://bz.apache.org/bugzilla/show_bug.cgi?id=53693 / https://www.mail-archive.com/dev@httpd.apache.org/msg55262.html
+            > apr_sleep(apr_time_from_msec(10));
+			> ap_log_rerror(APLOG_MARK, APLOG_WARNING, 0, r,"mod_fcgid: ### handle_request: wait 10ms");
             if (i > 0 || j > 0 || count_busy_processes(r, &fcgi_request)) {
                 apr_sleep(apr_time_from_sec(1));
 
@@ -459,9 +463,14 @@
                 if (bucket_ctx->procnode)
                     break;
             }
-
+*/
             /* Send a spawn request if I can't get a process slot */
             procmgr_send_spawn_cmd(&fcgi_request, r);
+
+			/* Try again */ 
+			bucket_ctx->procnode = apply_free_procnode(r, &fcgi_request); 
+			if (bucket_ctx->procnode) 
+				break; 
         }
 
         /* Connect to the fastcgi server */
