diff --git a/CMakeLists.txt b/CMakeLists.txt
index 28b9fbe..c07ecba 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -63,8 +63,8 @@ foreach(_build_type "Release" "MinSizeRel" "RelWithDebInfo")
   endforeach()
 endforeach()
 
-enable_testing()
-add_custom_target(check COMMAND ${CMAKE_CTEST_COMMAND})
+# enable_testing()
+# add_custom_target(check COMMAND ${CMAKE_CTEST_COMMAND})
 
 # Checks for header files.
 include(CheckIncludeFile)
@@ -185,8 +185,8 @@ set(PKGDATADIR "${CMAKE_INSTALL_FULL_DATADIR}/${CMAKE_PROJECT_NAME}")
 install(FILES README.rst DESTINATION "${CMAKE_INSTALL_DOCDIR}")
 
 add_subdirectory(lib)
-add_subdirectory(tests)
-add_subdirectory(examples)
+# add_subdirectory(tests)
+# add_subdirectory(examples)
 
 
 string(TOUPPER "${CMAKE_BUILD_TYPE}" _build_type)
diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index 3928fb3..ce21e4c 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -79,14 +79,14 @@ if(ENABLE_SHARED_LIB)
 endif()
 
 # Static library (for unittests because of symbol visibility)
-add_library(nghttp3_static STATIC ${nghttp3_SOURCES})
-set_target_properties(nghttp3_static PROPERTIES
-  COMPILE_FLAGS "${WARNCFLAGS}"
-  VERSION ${LT_VERSION} SOVERSION ${LT_SOVERSION}
-  ARCHIVE_OUTPUT_NAME nghttp3${STATIC_LIB_SUFFIX}
-  )
-target_compile_definitions(nghttp3_static PUBLIC "-DNGHTTP3_STATICLIB")
 if(ENABLE_STATIC_LIB)
+  add_library(nghttp3_static STATIC ${nghttp3_SOURCES})
+  set_target_properties(nghttp3_static PROPERTIES
+    COMPILE_FLAGS "${WARNCFLAGS}"
+    VERSION ${LT_VERSION} SOVERSION ${LT_SOVERSION}
+    ARCHIVE_OUTPUT_NAME nghttp3${STATIC_LIB_SUFFIX}
+    )
+  target_compile_definitions(nghttp3_static PUBLIC "-DNGHTTP3_STATICLIB")
   install(TARGETS nghttp3_static
     DESTINATION "${CMAKE_INSTALL_LIBDIR}")
 endif()
