diff --git a/Configure b/Configure
index 15054f9403..8ebcbd3dac 100755
--- a/Configure
+++ b/Configure
@@ -587,14 +587,11 @@ our %disabled = ( # "what"         => "comment"
                   "fips"                => "default",
                   "fips-jitter"         => "default",
                   "asan"                => "default",
-                  "brotli"              => "default",
-                  "brotli-dynamic"      => "default",
                   "buildtest-c++"       => "default",
                   "crypto-mdebug"       => "default",
                   "crypto-mdebug-backtrace" => "default",
                   "demos"               => "default",
                   "h3demo"              => "default",
-                  "hqinterop"           => "default",
                   "devcryptoeng"        => "default",
                   "ec_nistp_64_gcc_128" => "default",
                   "egd"                 => "default",
@@ -618,8 +615,6 @@ our %disabled = ( # "what"         => "comment"
                   "weak-ssl-ciphers"    => "default",
                   "zlib"                => "default",
                   "zlib-dynamic"        => "default",
-                  "zstd"                => "default",
-                  "zstd-dynamic"        => "default",
                 );
 
 # Note: => pair form used for aesthetics, not to truly make a hash table
