diff --git a/CMakeLists.txt b/CMakeLists.txt
index 201519c9..0426b7a0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1471,10 +1471,10 @@ file(
 file(GLOB man1 ${PROJECT_SOURCE_DIR}/doc/*.1)
 file(GLOB man3 ${PROJECT_SOURCE_DIR}/doc/*.3)
 
-install(FILES ${man1} DESTINATION ${CMAKE_INSTALL_MANDIR}/man1)
-install(FILES ${man3} DESTINATION ${CMAKE_INSTALL_MANDIR}/man3)
-install(FILES ${txts} DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/doc/pcre2)
-install(FILES ${html} DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/doc/pcre2/html)
+# install(FILES ${man1} DESTINATION ${CMAKE_INSTALL_MANDIR}/man1)
+# install(FILES ${man3} DESTINATION ${CMAKE_INSTALL_MANDIR}/man3)
+# install(FILES ${txts} DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/doc/pcre2)
+# install(FILES ${html} DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/doc/pcre2/html)
 
 if(MSVC AND INSTALL_MSVC_PDB)
   install(FILES ${DLL_PDB_FILES} DESTINATION ${CMAKE_INSTALL_BINDIR} CONFIGURATIONS RelWithDebInfo)
