diff --git "a/php7/memcache_pool.h" "b/php7/memcache_pool.h"
index 2382aa8..34497d0 100644
--- "a/php7/memcache_pool.h"
+++ "b/php7/memcache_pool.h"
@@ -415,15 +415,15 @@ ZEND_BEGIN_MODULE_GLOBALS(memcache)
 	long compress_threshold;
 	long lock_timeout;
 	char *session_key_prefix;
-	char *session_prefix_host_key;
-	char *session_prefix_host_key_remove_www;
-	char *session_prefix_host_key_remove_subdomain;
+	zend_bool session_prefix_host_key;
+	zend_bool session_prefix_host_key_remove_www;
+	zend_bool session_prefix_host_key_remove_subdomain;
 	char *session_prefix_static_key;
 	char *session_save_path;
 	char *key_prefix;
-	char *prefix_host_key;
-	char *prefix_host_key_remove_www;
-	char *prefix_host_key_remove_subdomain;
+	zend_bool prefix_host_key;
+	zend_bool prefix_host_key_remove_www;
+	zend_bool prefix_host_key_remove_subdomain;
 	char *prefix_static_key;
 ZEND_END_MODULE_GLOBALS(memcache)
 
