diff --git a/config.w32 b/config.w32
index dfc7551..2596550 100644
--- a/config.w32
+++ b/config.w32
@@ -314,7 +314,7 @@ function is_dev_mode() {
 }
 
 function is_makefile_frag_supported() {
-	return typeof PHP_MAKEFILE_FRAGMENTS !== typeof undefined;
+	return false; // typeof PHP_MAKEFILE_FRAGMENTS !== typeof undefined;
 }
 
 // ----------------------------------------------------------------------------
