diff --git a/ext/openssl/openssl.c b/ext/openssl/openssl.c
index 19e7a0d79e..c11ef36f29 100644
--- a/ext/openssl/openssl.c
+++ b/ext/openssl/openssl.c
@@ -1221,7 +1221,7 @@ PHP_MINIT_FUNCTION(openssl)
 	REGISTER_LONG_CONSTANT("OPENSSL_CMS_NOSIGS", CMS_NOSIGS, CONST_CS|CONST_PERSISTENT);
 
 	REGISTER_LONG_CONSTANT("OPENSSL_PKCS1_PADDING", RSA_PKCS1_PADDING, CONST_CS|CONST_PERSISTENT);
-	REGISTER_LONG_CONSTANT("OPENSSL_SSLV23_PADDING", RSA_SSLV23_PADDING, CONST_CS|CONST_PERSISTENT);
+//	REGISTER_LONG_CONSTANT("OPENSSL_SSLV23_PADDING", RSA_SSLV23_PADDING, CONST_CS|CONST_PERSISTENT);
 	REGISTER_LONG_CONSTANT("OPENSSL_NO_PADDING", RSA_NO_PADDING, CONST_CS|CONST_PERSISTENT);
 	REGISTER_LONG_CONSTANT("OPENSSL_PKCS1_OAEP_PADDING", RSA_PKCS1_OAEP_PADDING, CONST_CS|CONST_PERSISTENT);
 
diff --git a/main/network.c b/main/network.c
index 2c504952b2..708fe664e4 100644
--- a/main/network.c
+++ b/main/network.c
@@ -118,7 +118,9 @@ static const char *php_gai_strerror(int code)
                 {EAI_NONAME, "Name or service not known"},
                 {EAI_SERVICE, "Servname not supported for ai_socktype"},
                 {EAI_SOCKTYPE, "ai_socktype not supported"},
+#  ifdef EAI_SYSTEM
                 {EAI_SYSTEM, "System error"},
+#  endif
                 {0, NULL}
         };
         int i;
