diff --git a/ext/intl/breakiterator/codepointiterator_internal.cpp b/ext/intl/breakiterator/codepointiterator_internal.cpp
index 71ba056994..7ff8224ed3 100644
--- a/ext/intl/breakiterator/codepointiterator_internal.cpp
+++ b/ext/intl/breakiterator/codepointiterator_internal.cpp
@@ -73,7 +73,11 @@ CodePointBreakIterator::~CodePointBreakIterator()
 	clearCurrentCharIter();
 }
 
-UBool CodePointBreakIterator::operator==(const BreakIterator& that) const
+#if U_ICU_VERSION_MAJOR_NUM < 70
+	UBool CodePointBreakIterator::operator==(const BreakIterator& that) const
+#else
+	bool CodePointBreakIterator::operator==(const BreakIterator& that) const
+#endif
 {
 	if (typeid(*this) != typeid(that)) {
 		return false;
diff --git a/ext/intl/breakiterator/codepointiterator_internal.h b/ext/intl/breakiterator/codepointiterator_internal.h
index 43ec79d0b7..ad07c86a5d 100644
--- a/ext/intl/breakiterator/codepointiterator_internal.h
+++ b/ext/intl/breakiterator/codepointiterator_internal.h
@@ -37,7 +37,11 @@ namespace PHP {
 
 		virtual ~CodePointBreakIterator();
 
-		virtual UBool operator==(const BreakIterator& that) const;
+		#if U_ICU_VERSION_MAJOR_NUM < 70
+			virtual UBool operator==(const BreakIterator& that) const;
+		#else
+			virtual bool operator==(const BreakIterator& that) const;
+		#endif
 
 		virtual CodePointBreakIterator* clone(void) const;
 
diff --git a/main/network.c b/main/network.c
index 1ad7e370cc..4b7e9d628a 100644
--- a/main/network.c
+++ b/main/network.c
@@ -118,7 +118,9 @@ static const char *php_gai_strerror(int code)
 		{EAI_NONAME, "Name or service not known"},
 		{EAI_SERVICE, "Servname not supported for ai_socktype"},
 		{EAI_SOCKTYPE, "ai_socktype not supported"},
+#  ifdef EAI_SYSTEM
 		{EAI_SYSTEM, "System error"},
+#  endif
 		{0, NULL}
 	};
 	int i;
