diff --git "a/CMakeLists.txt" "b/CMakeLists.txt"
index 6a66091..46be120 100644
--- "a/CMakeLists.txt"
+++ "b/CMakeLists.txt"
@@ -46,10 +46,10 @@ set(CMAKE_MODULE_PATH "${SERF_SOURCE_DIR}/build")
 include(SerfVersion)
 
 project("Serf" VERSION ${SERF_VERSION} LANGUAGES C)
-message(WARNING
-        "Serf's CMake build is considered EXPERIMENTAL. "
-        "Some features are not supported and the build "
-        "has not been tested on many supported platforms.")
+# message(WARNING
+#        "Serf's CMake build is considered EXPERIMENTAL. "
+#        "Some features are not supported and the build "
+#        "has not been tested on many supported platforms.")
 
 
 # Build options
