diff --git "a/CMakeLists.txt" "b/CMakeLists.txt"
index 405d654..c14620a 100644
--- "a/CMakeLists.txt"
+++ "b/CMakeLists.txt"
@@ -425,7 +425,7 @@ else()
 endif()
 
 message(STATUS "Build shared library:                                 ${ENABLE_SHARED}")
-message(STATUS "Shared library name:                                  ${CMAKE_SHARED_LIBRARY_PREFIX}${PROJECT_NAME}")
+message(STATUS "Shared library name:                                  ${SHAREDLIB_PREFIX}${PROJECT_NAME}")
 message(STATUS "Build static library:                                 ${ENABLE_STATIC}")
 message(STATUS "Build shell tool:                                     ${BUILD_SHELL}")
 message(STATUS "Statically link shell tool:                           ${ENABLE_STATIC_SHELL}")
@@ -489,7 +489,7 @@ if(ENABLE_SHARED)
     target_link_libraries(sqlite3-shared ${SQLITE3_DEP_LIBRARIES})
     target_include_directories(sqlite3-shared PUBLIC ${CMAKE_INSTALL_PREFIX}/include)
     set_target_properties(sqlite3-shared PROPERTIES
-        OUTPUT_NAME ${PROJECT_NAME}
+        OUTPUT_NAME ${SHAREDLIB_PREFIX}${PROJECT_NAME}
         VERSION ${PROJECT_VERSION}
         SOVERSION ${PROJECT_VERSION_MAJOR})
 
