diff --git a/common.c b/common.c
index bc086e5..82dd173 100644
--- a/common.c
+++ b/common.c
@@ -25,12 +25,6 @@
 #include <sys/prctl.h>
 #endif
 
-/* Added to make the code compilable under CYGWIN
- * */
-#ifndef SA_NOCLDWAIT
-#define SA_NOCLDWAIT 0
-#endif
-
 /* Make use of systemd socket activation
  * */
 #ifdef SYSTEMD
diff --git a/common.h b/common.h
index e56bd1e..80766c5 100644
--- a/common.h
+++ b/common.h
@@ -84,6 +84,12 @@
 #define TCP_FASTOPEN_CONNECT 30 /* Attempt FastOpen with connect.  */
 #endif
 
+/* Added to make the code compilable under CYGWIN
+ * */
+#ifndef SA_NOCLDWAIT
+#define SA_NOCLDWAIT 0
+#endif
+
 enum connection_state {
     ST_PROBING=1,    /* Waiting for timeout to find where to forward */
     ST_SHOVELING   /* Connexion is established */
