diff --git "a/build/generator/gen_win_dependencies.py" "b/build/generator/gen_win_dependencies.py"
index 4b254c90d3..d0f4868e82 100644
--- "a/build/generator/gen_win_dependencies.py"
+++ "b/build/generator/gen_win_dependencies.py"
@@ -1147,7 +1147,7 @@ class GenDependenciesBase(gen_base.GeneratorBase):
                                stderr=subprocess.STDOUT).stdout
       line = outfp.read().decode('utf8')
       if line:
-        vermatch = re.search(r'(([0-9]+(\.[0-9]+)+)(_[._0-9]+)?)', line, re.M)
+        vermatch = re.search(r'(([0-9]+((\.[0-9]+)+)?)(_[._0-9]+)?)', line, re.M)
       else:
         vermatch = None
 
