 config.w32 | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/config.w32 b/config.w32
index d0b82ff7..e1753f67 100644
--- a/config.w32
+++ b/config.w32
@@ -17,13 +17,13 @@ if (PHP_XDEBUG != 'no') {
 	
 	var files = "xdebug.c";
 
-	var XDEBUG_PHP_VERSION = 10000 * PHP_VERSION + 100 * PHP_MINOR_VERSION + PHP_RELEASE_VERSION;
+	var XDEBUG_PHP_VERSION = 10000 * PHP_VERSION + 100 * PHP_MINOR_VERSION + 1 * PHP_RELEASE_VERSION;
 	if (XDEBUG_PHP_VERSION < 80000) {
 		ERROR("not supported. Need a PHP version >= 8.0.0 and < 8.5.0 (found " + XDEBUG_PHP_VERSION + ")");
 	} else if (XDEBUG_PHP_VERSION >= 80500) {
 		ERROR("not supported. Need a PHP version >= 8.0.0 and < 8.5.0 (found " + XDEBUG_PHP_VERSION + ")");
 	} else {
-		MESSAGE("supported (" + XDEBUG_PHP_VERSION + ")");
+		// MESSAGE("supported (" + XDEBUG_PHP_VERSION + ")");
 	}
 
 	if (typeof(ZEND_EXTENSION) == 'undefined') {
