From c8a29353df6dd244b58b3bb938795ea2ce699333 Mon Sep 17 00:00:00 2001
From: Derick Rethans <github@derickrethans.nl>
Date: Wed, 22 Oct 2025 10:10:09 +0100
Subject: [PATCH 1/4] Build Xdebug 3.5 on PHP 8.5 on Windows in CI

---
 .github/workflows/tests.yml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/.github/workflows/tests.yml b/.github/workflows/tests.yml
index 2e9f4bc7a..3001a3b2f 100644
--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -23,7 +23,7 @@ jobs:
               id: extension-matrix
               uses: php/php-windows-builder/extension-matrix@v1
               with:
-                php-version-list: '8.0, 8.1, 8.2, 8.3, 8.4'
+                php-version-list: '8.0, 8.1, 8.2, 8.3, 8.4, 8.5'
                 arch-list: 'x64'
 
     windows:

From 66ceff9a1d7b8c1db4df034a98a4b9fcd2471e2a Mon Sep 17 00:00:00 2001
From: Derick Rethans <github@derickrethans.nl>
Date: Wed, 22 Oct 2025 11:15:53 +0100
Subject: [PATCH 2/4] Fixed a memory leak while mapping paths

---
 src/lib/maps/maps_private.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/lib/maps/maps_private.c b/src/lib/maps/maps_private.c
index dfd0ed5cf..a5206a1ec 100644
--- a/src/lib/maps/maps_private.c
+++ b/src/lib/maps/maps_private.c
@@ -131,6 +131,8 @@ int remote_to_local(xdebug_path_maps *maps, const char *remote_path, size_t remo
 		case XDEBUG_PATH_MAP_TYPE_FILE:
 			*local_path = xdebug_str_copy(result->m.local_path);
 			*local_line = remote_line;
+
+			xdfree(url_path);
 			return result->type;
 
 		case XDEBUG_PATH_MAP_TYPE_LINES: {
@@ -280,6 +282,8 @@ int local_to_remote(xdebug_path_maps *maps, const char *local_path, size_t local
 			size_t      result_line;
 
 			if (!find_remote_line_number_from_ranges(local_line, result->m.line_ranges, &result_path, &result_line)) {
+				xdfree(url_path);
+
 				return XDEBUG_PATH_MAP_TYPE_UNKNOWN;
 			}
 

From c461a58b80db38c2c3fef93fb3d277d1c1d2374d Mon Sep 17 00:00:00 2001
From: Derick Rethans <github@derickrethans.nl>
Date: Wed, 22 Oct 2025 11:18:48 +0100
Subject: [PATCH 3/4] Split xdebug_normalize_path_* into their own file

---
 config.m4                   |  2 +-
 config.w32                  |  2 +-
 package.xml                 |  2 ++
 src/lib/maps/maps_private.c |  2 +-
 src/lib/maps/parser.c       |  2 +-
 src/lib/normalize_path.c    | 54 +++++++++++++++++++++++++++++++++++++
 src/lib/normalize_path.h    | 35 ++++++++++++++++++++++++
 src/lib/usefulstuff.c       | 34 -----------------------
 src/lib/usefulstuff.h       |  7 -----
 tests/ctest/Makefile        | 13 +++++----
 10 files changed, 103 insertions(+), 50 deletions(-)
 create mode 100644 src/lib/normalize_path.c
 create mode 100644 src/lib/normalize_path.h

diff --git a/config.m4 b/config.m4
index dd1c22b71..2ade7704b 100644
--- a/config.m4
+++ b/config.m4
@@ -108,7 +108,7 @@ if test "$PHP_XDEBUG" != "no"; then
   PHP_XDEBUG_CFLAGS="$STD_CFLAGS $MAINTAINER_CFLAGS"
 
   XDEBUG_BASE_SOURCES="src/base/base.c src/base/ctrl_socket.c src/base/filter.c"
-  XDEBUG_LIB_SOURCES="src/lib/usefulstuff.c src/lib/cmd_parser.c src/lib/compat.c src/lib/crc32.c src/lib/file.c src/lib/hash.c src/lib/headers.c src/lib/lib.c src/lib/llist.c src/lib/log.c src/lib/set.c src/lib/str.c src/lib/timing.c src/lib/trim.c src/lib/var.c src/lib/var_export_html.c src/lib/var_export_line.c src/lib/var_export_text.c src/lib/var_export_xml.c src/lib/xdebug_strndup.c src/lib/xml.c"
+  XDEBUG_LIB_SOURCES="src/lib/usefulstuff.c src/lib/cmd_parser.c src/lib/compat.c src/lib/crc32.c src/lib/file.c src/lib/hash.c src/lib/headers.c src/lib/lib.c src/lib/llist.c src/lib/log.c src/lib/normalize_path.c src/lib/set.c src/lib/str.c src/lib/timing.c src/lib/trim.c src/lib/var.c src/lib/var_export_html.c src/lib/var_export_line.c src/lib/var_export_text.c src/lib/var_export_xml.c src/lib/xdebug_strndup.c src/lib/xml.c"
   XDEBUG_LIB_MAPS_SOURCES="src/lib/maps/maps.c src/lib/maps/maps_private.c src/lib/maps/parser.c"
 
   XDEBUG_COVERAGE_SOURCES="src/coverage/branch_info.c src/coverage/code_coverage.c"
diff --git a/config.w32 b/config.w32
index ed9bb3b7a..43f913f29 100644
--- a/config.w32
+++ b/config.w32
@@ -6,7 +6,7 @@ ARG_WITH("xdebug-compression", "whether to compress profiler files (requires zli
 
 if (PHP_XDEBUG != 'no') {
 	var XDEBUG_BASE_SOURCES="base.c filter.c ctrl_socket.c"
-	var XDEBUG_LIB_SOURCES="usefulstuff.c cmd_parser.c compat.c crc32.c file.c hash.c headers.c lib.c llist.c log.c set.c str.c timing.c trim.c var.c var_export_html.c var_export_line.c var_export_text.c var_export_xml.c xdebug_strndup.c xml.c"
+	var XDEBUG_LIB_SOURCES="usefulstuff.c cmd_parser.c compat.c crc32.c file.c hash.c headers.c lib.c llist.c log.c normalize_path.c set.c str.c timing.c trim.c var.c var_export_html.c var_export_line.c var_export_text.c var_export_xml.c xdebug_strndup.c xml.c"
 	var XDEBUG_LIB_MAPS_SOURCES="maps.c maps_private.c parser.c"
 
 	var XDEBUG_COVERAGE_SOURCES="branch_info.c code_coverage.c"
diff --git a/package.xml b/package.xml
index 252af6341..3f39ada6c 100644
--- a/package.xml
+++ b/package.xml
@@ -133,6 +133,8 @@ Tue, Oct 07, 2025 - Xdebug 3.5.0alpha2
      <file name="llist.c" role="src" />
      <file name="llist.h" role="src" />
      <file name="mm.h" role="src" />
+     <file name="normalize_path.c" role="src" />
+     <file name="normalize_path.h" role="src" />
      <file name="php-header.h" role="src" />
      <file name="set.c" role="src" />
      <file name="set.h" role="src" />
diff --git a/src/lib/maps/maps_private.c b/src/lib/maps/maps_private.c
index a5206a1ec..1e5da112c 100644
--- a/src/lib/maps/maps_private.c
+++ b/src/lib/maps/maps_private.c
@@ -19,8 +19,8 @@
 
 #include "maps_private.h"
 #include "../mm.h"
+#include "../normalize_path.h"
 #include "../strrnchr.h"
-#include "../usefulstuff.h"
 #include "../vector.h"
 #include "../xdebug_strndup.h"
 
diff --git a/src/lib/maps/parser.c b/src/lib/maps/parser.c
index b6e3d2835..8adedba3c 100644
--- a/src/lib/maps/parser.c
+++ b/src/lib/maps/parser.c
@@ -25,9 +25,9 @@
 
 #include "../hash.h"
 #include "../mm.h"
+#include "../normalize_path.h"
 #include "../str.h"
 #include "../trim.h"
-#include "../usefulstuff.h"
 #include "../xdebug_strndup.h"
 
 typedef struct path_maps_parser_state {
diff --git a/src/lib/normalize_path.c b/src/lib/normalize_path.c
new file mode 100644
index 000000000..8aca55a51
--- /dev/null
+++ b/src/lib/normalize_path.c
@@ -0,0 +1,54 @@
+/*
+   +----------------------------------------------------------------------+
+   | Xdebug                                                               |
+   +----------------------------------------------------------------------+
+   | Copyright (c) 2002-2025 Derick Rethans                               |
+   +----------------------------------------------------------------------+
+   | This source file is subject to version 1.01 of the Xdebug license,   |
+   | that is bundled with this package in the file LICENSE, and is        |
+   | available at through the world-wide-web at                           |
+   | https://xdebug.org/license.php                                       |
+   | If you did not receive a copy of the Xdebug license and are unable   |
+   | to obtain it through the world-wide-web, please send a note to       |
+   | derick@xdebug.org so we can mail you a copy immediately.             |
+   +----------------------------------------------------------------------+
+ */
+
+#include <string.h>
+
+#include "mm.h"
+#include "str.h"
+
+#ifdef PHP_WIN32
+char *xdebug_normalize_path_char(const char *path)
+{
+	char *new_path = xdstrdup(path);
+	char *ptr = new_path;
+
+	do {
+		if ((*ptr) == '\\') {
+			*ptr = '/';
+		}
+		ptr++;
+	} while (*ptr != '\0');
+
+	return new_path;
+}
+
+void xdebug_normalize_path_xdebug_str_in_place(xdebug_str *path)
+{
+	int i;
+
+	for (i = 0; i < XDEBUG_STR_LEN(path); i++) {
+		if (XDEBUG_STR_VAL(path)[i] == '\\') {
+			XDEBUG_STR_VAL(path)[i] = '/';
+		}
+	}
+}
+#else
+char *xdebug_normalize_path_char(const char *path)
+{
+	return xdstrdup(path);
+}
+#endif
+
diff --git a/src/lib/normalize_path.h b/src/lib/normalize_path.h
new file mode 100644
index 000000000..16e819ab8
--- /dev/null
+++ b/src/lib/normalize_path.h
@@ -0,0 +1,35 @@
+/*
+   +----------------------------------------------------------------------+
+   | Xdebug                                                               |
+   +----------------------------------------------------------------------+
+   | Copyright (c) 2002-2025 Derick Rethans                               |
+   +----------------------------------------------------------------------+
+   | This source file is subject to version 1.01 of the Xdebug license,   |
+   | that is bundled with this package in the file LICENSE, and is        |
+   | available at through the world-wide-web at                           |
+   | https://xdebug.org/license.php                                       |
+   | If you did not receive a copy of the Xdebug license and are unable   |
+   | to obtain it through the world-wide-web, please send a note to       |
+   | derick@xdebug.org so we can mail you a copy immediately.             |
+   +----------------------------------------------------------------------+
+ */
+
+#ifndef __HAVE_NORMALIZE_PATH_H__
+#define __HAVE_NORMALIZE_PATH_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+char *xdebug_normalize_path_char(const char *path);
+#ifdef PHP_WIN32
+void xdebug_normalize_path_xdebug_str_in_place(xdebug_str *path);
+#else
+# define xdebug_normalize_path_xdebug_str_in_place(path)
+#endif
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif
diff --git a/src/lib/usefulstuff.c b/src/lib/usefulstuff.c
index 27a9c2dc0..9a73e343c 100644
--- a/src/lib/usefulstuff.c
+++ b/src/lib/usefulstuff.c
@@ -722,37 +722,3 @@ int xdebug_format_filename(char **formatted_name, const char *default_fmt, zend_
 
 	return fname.l;
 }
-
-
-#ifdef PHP_WIN32
-char *xdebug_normalize_path_char(const char *path)
-{
-	char *new_path = xdstrdup(path);
-	char *ptr = new_path;
-
-	do {
-		if ((*ptr) == '\\') {
-			*ptr = '/';
-		}
-		ptr++;
-	} while (*ptr != '\0');
-
-	return new_path;
-}
-
-void xdebug_normalize_path_xdebug_str_in_place(xdebug_str *path)
-{
-	int i;
-
-	for (i = 0; i < XDEBUG_STR_LEN(path); i++) {
-		if (XDEBUG_STR_VAL(path)[i] == '\\') {
-			XDEBUG_STR_VAL(path)[i] = '/';
-		}
-	}
-}
-#else
-char *xdebug_normalize_path_char(const char *path)
-{
-	return xdstrdup(path);
-}
-#endif
diff --git a/src/lib/usefulstuff.h b/src/lib/usefulstuff.h
index 6528f7eb5..a51d01bdf 100644
--- a/src/lib/usefulstuff.h
+++ b/src/lib/usefulstuff.h
@@ -38,13 +38,6 @@ char *xdebug_zstr_path_to_url(zend_string *string);
 char *xdebug_xdebug_str_path_to_url(xdebug_str *string);
 char *xdebug_path_from_url(zend_string *fileurl);
 
-char *xdebug_normalize_path_char(const char *path);
-#ifdef PHP_WIN32
-void xdebug_normalize_path_xdebug_str_in_place(xdebug_str *path);
-#else
-# define xdebug_normalize_path_xdebug_str_in_place(path)
-#endif
-
 FILE *xdebug_fopen(char *fname, const char *mode, const char *extension, char **new_fname);
 int xdebug_format_output_filename(char **filename, char *format, char *script_name);
 int xdebug_format_file_link(char **filename, const char *error_filename, int error_lineno);
diff --git a/tests/ctest/Makefile b/tests/ctest/Makefile
index 10a04171e..0e79d376c 100644
--- a/tests/ctest/Makefile
+++ b/tests/ctest/Makefile
@@ -23,14 +23,17 @@ afl-test-parser: xdebuglib.a ${C_TESTS} ${SOURCE_FILES} ${HEADER_FILES} fuzzing-
 afl-test-mapper: xdebuglib.a ${C_TESTS} ${SOURCE_FILES} ${HEADER_FILES} fuzzing-shim-mapper.c
 	afl-g++ ${CPPFLAGS} fuzzing-shim-mapper.c ${SOURCE_FILES} xdebuglib.a -I ${INCLUDE_PATH} -o afl-test-mapper ${LDFLAGS}
 
-xdebuglib.a: hash.o llist.o str.o trim.o xdebug_strndup.o
-	ar -rc xdebuglib.a hash.o llist.o str.o trim.o xdebug_strndup.o
+xdebuglib.a: hash.o llist.o normalize_path.o str.o trim.o xdebug_strndup.o
+	ar -rc xdebuglib.a hash.o llist.o normalize_path.o str.o trim.o xdebug_strndup.o
+
+hash.o: ../../src/lib/hash.c
+	gcc -c ${CPPFLAGS} -o hash.o ../../src/lib/hash.c
 
 llist.o: ../../src/lib/llist.c
 	gcc -c ${CPPFLAGS} -o llist.o ../../src/lib/llist.c
 
-hash.o: ../../src/lib/hash.c
-	gcc -c ${CPPFLAGS} -o hash.o ../../src/lib/hash.c
+normalize_path.o: ../../src/lib/normalize_path.c
+	gcc -c ${CPPFLAGS} -o normalize_path.o ../../src/lib/normalize_path.c
 
 str.o: ../../src/lib/str.c
 	gcc -c ${CPPFLAGS} -o str.o ../../src/lib/str.c
@@ -42,4 +45,4 @@ xdebug_strndup.o: ../../src/lib/xdebug_strndup.c
 	gcc -c ${CPPFLAGS} -o xdebug_strndup.o ../../src/lib/xdebug_strndup.c
 
 clean:
-	rm xdebuglib.a hash.o llist.o str.o trim.o xdebug_strndup.o ctest afl-test-parser afl-test-mapper
+	rm xdebuglib.a hash.o llist.o normalize_path.o str.o trim.o xdebug_strndup.o ctest afl-test-parser afl-test-mapper

From b8bc9425390ca7745abdc715a359d2f369c1a07b Mon Sep 17 00:00:00 2001
From: Derick Rethans <github@derickrethans.nl>
Date: Wed, 22 Oct 2025 11:19:39 +0100
Subject: [PATCH 4/4] Fixed issue #2372: PHP 8.5: Changes to glob interface

---
 src/lib/maps/maps.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/src/lib/maps/maps.c b/src/lib/maps/maps.c
index 560dff7eb..a96777c73 100644
--- a/src/lib/maps/maps.c
+++ b/src/lib/maps/maps.c
@@ -16,6 +16,10 @@
 #include <stddef.h>
 #include <stdlib.h>
 
+#ifndef XDEBUG_NO_PHP_FEATURES
+# include "php_xdebug.h"
+#endif
+
 #if PHP_VERSION_ID >= 80500
 # include "php_glob.h"
 # define xdebug_glob php_glob
@@ -33,7 +37,9 @@
 # define xdebug_globfree globfree
 #endif
 
-#include "php_xdebug.h"
+#ifndef PHP_GLOB_NOMATCH
+# define PHP_GLOB_NOMATCH GLOB_NOMATCH
+#endif
 
 #include "maps_private.h"
 #include "parser.h"
@@ -85,7 +91,7 @@ static void scan_directory(const char *dir)
 		case 0: /* No error */
 			break;
 
-		case GLOB_NOMATCH:
+		case PHP_GLOB_NOMATCH:
 			xdebug_log_ex(XLOG_CHAN_PATHMAP, XLOG_DEBUG, "NOMATCH", "No map files found with pattern '%s'", scan_dir);
 			xdfree(scan_dir);
 
