diff --git a/Zend/zend_call_stack.c b/Zend/zend_call_stack.c
index ac12b8b7d87..32edffdb6ee 100644
--- a/Zend/zend_call_stack.c
+++ b/Zend/zend_call_stack.c
@@ -416,7 +416,12 @@ static bool zend_call_stack_get_win32(zend_call_stack *stack)
 	 *                v  Lower addresses   v
 	 */
 
-	GetCurrentThreadStackLimits(&low_limit, &high_limit);
+ 	NT_TIB* tib = (NT_TIB*)NtCurrentTeb();
+ 	high_limit = (ULONG_PTR)tib->StackBase;
+ 	MEMORY_BASIC_INFORMATION mbi;
+ 	if (VirtualQuery(tib->StackLimit, &mbi, sizeof(mbi))) {
+ 		low_limit = (ULONG_PTR)mbi.AllocationBase;
+ 	}
 
 	result_size = VirtualQuery((void*)low_limit,
 			&uncommitted_region, sizeof(uncommitted_region));
diff --git a/win32/time.c b/win32/time.c
index ac1abbc9e9c..f0575ce886e 100644
--- a/win32/time.c
+++ b/win32/time.c
@@ -29,7 +29,7 @@ static zend_always_inline void getfilesystemtime(struct timeval *tv)
 	unsigned __int64 ff = 0;
 	ULARGE_INTEGER fft;
 
-	GetSystemTimePreciseAsFileTime(&ft);
+	GetSystemTimeAsFileTime(&ft);
 
 	/*
 	 * Do not cast a pointer to a FILETIME structure to either a
