From 648449175f40ba0917f181e5521a5612644aa91c Mon Sep 17 00:00:00 2001
From: Kai Pastor <dg0yt@darc.de>
Date: Sun, 30 Oct 2022 10:40:59 +0100
Subject: [PATCH] Use libwebp pc module instead of libs

---
 CMakeLists.txt     | 1 +
 src/CMakeLists.txt | 1 -
 2 files changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0f488518e..dca3fbe4c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -218,6 +218,7 @@ else (USE_EXT_GD)
 	IF(WEBP_FOUND)
 		INCLUDE_DIRECTORIES(${WEBP_INCLUDE_DIR})
 		SET(HAVE_LIBWEBP 1)
+		LIST(APPEND PKG_REQUIRES_PRIVATES libwebp)
 	ENDIF(WEBP_FOUND)
 
 	IF(HEIF_FOUND)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 1069c4f77..23c94fe85 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -138,7 +138,6 @@ endif()
 SET(LIBS_PRIVATES
 	${ICONV_LIBRARIES}
 	${LIQ_LIBRARIES}
-	${WEBP_LIBRARIES}
 )
 
 set(GD_PROGRAMS gdcmpgif)
