From baf3e3bfc44fc42d466c0ed950ed4951ee344bc8 Mon Sep 17 00:00:00 2001
From: David Carlier <devnexen@gmail.com>
Date: Wed, 10 Sep 2025 21:03:11 +0100
Subject: [PATCH] gdImageRectangle() integer overflow fix.

---
 src/gd.c | 26 +++++++++++++++++++++-----
 1 file changed, 21 insertions(+), 5 deletions(-)

diff --git a/src/gd.c b/src/gd.c
index 768fd2065..930a9496a 100644
--- a/src/gd.c
+++ b/src/gd.c
@@ -1183,12 +1183,16 @@ BGD_DECLARE(int) gdImageColorReplaceCallback (gdImagePtr im, gdCallbackImageColo
 static int
 clip_1d (int *x0, int *y0, int *x1, int *y1, int mindim, int maxdim)
 {
-	double m;			/* gradient of line */
+	double m, tmp;			/* gradient of line */
 	if (*x0 < mindim) {
 		/* start of line is left of window */
 		if (*x1 < mindim)		/* as is the end, so the line never cuts the window */
 			return 0;
-		m = (*y1 - *y0) / (double) (*x1 - *x0);	/* calculate the slope of the line */
+		tmp = (double)*x1 - (double)*x0;
+		if (tmp < (double)INT_MIN || tmp == 0.0 || tmp > (double)INT_MAX) {
+			return 0;
+		}
+		m = (*y1 - *y0) / tmp;	/* calculate the slope of the line */
 		/* adjust x0 to be on the left boundary (ie to be zero), and y0 to match */
 		*y0 -= (int)(m * (*x0 - mindim));
 		*x0 = mindim;
@@ -1204,7 +1208,11 @@ clip_1d (int *x0, int *y0, int *x1, int *y1, int mindim, int maxdim)
 		complement of above */
 		if (*x1 > maxdim)		/* as is the end, so the line misses the window */
 			return 0;
-		m = (*y1 - *y0) / (double) (*x1 - *x0);	/* calculate the slope of the line */
+		tmp = (double)*x1 - (double)*x0;
+		if (tmp < (double)INT_MIN || tmp == 0.0 || tmp > (double)INT_MAX) {
+			return 0;
+		}
+		m = (*y1 - *y0) / tmp;	/* calculate the slope of the line */
 		*y0 += (int)(m * (maxdim - *x0));	/* adjust so point is on the right
 							   boundary */
 		*x0 = maxdim;
@@ -1218,14 +1226,22 @@ clip_1d (int *x0, int *y0, int *x1, int *y1, int mindim, int maxdim)
 	/* the final case - the start of the line is inside the window */
 	if (*x1 > maxdim) {
 		/* other end is outside to the right */
-		m = (*y1 - *y0) / (double) (*x1 - *x0);	/* calculate the slope of the line */
+		tmp = (double)*x1 - (double)*x0;
+		if (tmp < (double)INT_MIN || tmp == 0.0 || tmp > (double)INT_MAX) {
+			return 0;
+		}
+		m = (*y1 - *y0) / tmp;	/* calculate the slope of the line */
 		*y1 += (int)(m * (maxdim - *x1));
 		*x1 = maxdim;
 		return 1;
 	}
 	if (*x1 < mindim) {
 		/* other end is outside to the left */
-		m = (*y1 - *y0) / (double) (*x1 - *x0);	/* calculate the slope of the line */
+		tmp = (double)*x1 - (double)*x0;
+		if (tmp < (double)INT_MIN || tmp == 0.0 || tmp > (double)INT_MAX) {
+			return 0;
+		}
+		m = (*y1 - *y0) / tmp;	/* calculate the slope of the line */
 		*y1 -= (int)(m * (*x1 - mindim));
 		*x1 = mindim;
 		return 1;
