diff --git "a/c:\\temp\\TortoiseGit\\CMakeLists-d495914.000.txt" "b/C:\\src\\httpd\\CMakeLists.txt"
index 3cd8a34d14..e0c7fd8aa6 100644
--- "a/c:\\temp\\TortoiseGit\\CMakeLists-d495914.000.txt"
+++ "b/C:\\src\\httpd\\CMakeLists.txt"
@@ -26,6 +26,7 @@ FIND_PACKAGE(LibXml2)
 FIND_PACKAGE(Lua51)
 FIND_PACKAGE(OpenSSL)
 FIND_PACKAGE(ZLIB)
+FIND_PACKAGE(CURL)
 
 # Options for support libraries not supported by cmake-bundled FindFOO
 
@@ -202,6 +203,7 @@ ENDIF()
 MESSAGE(STATUS "")
 MESSAGE(STATUS "Summary of feature detection:")
 MESSAGE(STATUS "")
+MESSAGE(STATUS "CURL_FOUND ............... : ${CURL_FOUND}")
 MESSAGE(STATUS "LIBXML2_FOUND ............ : ${LIBXML2_FOUND}")
 MESSAGE(STATUS "LUA51_FOUND .............. : ${LUA51_FOUND}")
 MESSAGE(STATUS "NGHTTP2_FOUND ............ : ${NGHTTP2_FOUND}")
@@ -351,6 +353,8 @@ SET(MODULE_LIST
   "modules/slotmem/mod_slotmem_plain+I+slotmem provider that uses plain memory"
   "modules/slotmem/mod_slotmem_shm+I+slotmem provider that uses shared memory"
   "modules/ssl/mod_ssl+i+SSL/TLS support"
+  "modules/md/mod_md+i+certificate provisioning via the ACME protocol"
+  "modules/fcgid/mod_fcgid+i+FastCGI implementation for Apache HTTP"
   "modules/test/mod_dialup+O+rate limits static files to dialup modem speeds"
   "modules/test/mod_optional_fn_export+O+example optional function exporter"
   "modules/test/mod_optional_fn_import+O+example optional function importer"
@@ -497,6 +501,31 @@ SET(mod_ssl_extra_sources
   modules/ssl/ssl_util.c             modules/ssl/ssl_util_ocsp.c
   modules/ssl/ssl_util_ssl.c         modules/ssl/ssl_util_stapling.c
 )
+IF(CURL_FOUND)
+  SET(mod_md_extra_includes           "${CURL_INCLUDE_DIR};${OPENSSL_INCLUDE_DIR};${JANSSON_INCLUDE_DIR}")
+  SET(mod_md_extra_libs               "${CURL_LIBRARIES};${OPENSSL_LIBRARIES};${JANSSON_LIBRARIES}")
+ENDIF()
+SET(mod_md_extra_sources
+  modules/md/md_acme.c		  modules/md/md_acme_acct.c
+  modules/md/md_acme_authz.c	  modules/md/md_acme_drive.c
+  modules/md/md_cmd_acme.c	  modules/md/md_cmd_main.c
+  modules/md/md_cmd_reg.c	  modules/md/md_cmd_store.c
+  modules/md/md_core.c		  modules/md/md_crypt.c
+  modules/md/md_curl.c		  modules/md/md_http.c
+  modules/md/md_json.c		  modules/md/md_jws.c
+  modules/md/md_log.c		  modules/md/md_reg.c
+  modules/md/md_store.c		  modules/md/md_store_fs.c
+  modules/md/md_util.c		  modules/md/mod_md.c
+  modules/md/mod_md_config.c	  modules/md/mod_md_os.c
+)
+SET(mod_fcgid_extra_sources
+modules/fcgid/fcgid_proc_win.c		modules/fcgid/fcgid_proctbl_win.c
+modules/fcgid/fcgid_protocol.c		modules/fcgid/fcgid_spawn_ctl.c
+modules/fcgid/mod_fcgid.c		modules/fcgid/fcgid_bridge.c
+modules/fcgid/fcgid_bucket.c		modules/fcgid/fcgid_conf.c
+modules/fcgid/fcgid_filter.c		modules/fcgid/fcgid_pm_main.c
+modules/fcgid/fcgid_pm_win.c		
+)
 SET(mod_status_extra_defines         STATUS_DECLARE_EXPORT)
 SET(mod_watchdog_install_lib 1)
 SET(mod_xml2enc_requires             LIBXML2_FOUND)
@@ -564,7 +593,7 @@ ELSE()
 ENDIF()
 
 # for easy reference from .dll/.so builds
-CONFIGURE_FILE(os/win32/BaseAddr.ref ${PROJECT_BINARY_DIR}/ COPYONLY)
+# CONFIGURE_FILE(os/win32/BaseAddr.ref ${PROJECT_BINARY_DIR}/ COPYONLY)
 
 ADD_EXECUTABLE(gen_test_char server/gen_test_char.c)
 GET_TARGET_PROPERTY(GEN_TEST_CHAR_EXE gen_test_char LOCATION)
@@ -636,6 +665,8 @@ SET(LIBHTTPD_SOURCES
 
 CONFIGURE_FILE(os/win32/win32_config_layout.h
                ${PROJECT_BINARY_DIR}/ap_config_layout.h)
+CONFIGURE_FILE(server/mpm/winnt/mpm_winnt.h
+               ${PROJECT_BINARY_DIR}/mpm_winnt.h)
 
 SET(HTTPD_INCLUDE_DIRECTORIES
   ${PROJECT_BINARY_DIR}
@@ -651,6 +682,8 @@ SET(HTTPD_INCLUDE_DIRECTORIES
   ${CMAKE_CURRENT_SOURCE_DIR}/modules/proxy
   ${CMAKE_CURRENT_SOURCE_DIR}/modules/session
   ${CMAKE_CURRENT_SOURCE_DIR}/modules/ssl
+  ${CMAKE_CURRENT_SOURCE_DIR}/modules/md
+  ${CMAKE_CURRENT_SOURCE_DIR}/modules/fcgid
   ${CMAKE_CURRENT_SOURCE_DIR}/server
   ${APR_INCLUDE_DIR}
   ${PCRE_INCLUDE_DIR}
@@ -677,6 +710,7 @@ SET(other_installed_h
   ${CMAKE_CURRENT_SOURCE_DIR}/modules/session/mod_session.h
   ${CMAKE_CURRENT_SOURCE_DIR}/modules/ssl/mod_ssl.h
   ${CMAKE_CURRENT_SOURCE_DIR}/modules/ssl/mod_ssl_openssl.h
+  ${CMAKE_CURRENT_SOURCE_DIR}/modules/md/mod_md.h
 )
 # When mod_serf is buildable, don't forget to copy modules/proxy/mod_serf.h
 
@@ -765,7 +799,7 @@ FOREACH (mod ${MODULE_PATHS})
     SET(mod_extra_libs "${mod_name}_extra_libs")
     SET_TARGET_PROPERTIES(${mod_name} PROPERTIES
       SUFFIX .so
-      LINK_FLAGS /base:@${PROJECT_BINARY_DIR}/BaseAddr.ref,${mod_name}.so
+      # LINK_FLAGS /base:@${PROJECT_BINARY_DIR}/BaseAddr.ref,${mod_name}.so
     )
     TARGET_LINK_LIBRARIES(${mod_name} ${${mod_extra_libs}} libhttpd ${EXTRA_LIBS} ${APR_LIBRARIES} ${HTTPD_SYSTEM_LIBS})
     DEFINE_WITH_BLANKS(define_long_name "LONG_NAME" "${mod_name} for Apache HTTP Server")
@@ -790,9 +824,9 @@ ENDFOREACH()
 
 ###########   HTTPD LIBRARIES   ############
 ADD_LIBRARY(libhttpd SHARED ${LIBHTTPD_SOURCES} build/win32/httpd.rc)
-SET_TARGET_PROPERTIES(libhttpd PROPERTIES
-  LINK_FLAGS /base:@${PROJECT_BINARY_DIR}/BaseAddr.ref,libhttpd.dll
-)
+#SET_TARGET_PROPERTIES(libhttpd PROPERTIES
+#  LINK_FLAGS /base:@${PROJECT_BINARY_DIR}/BaseAddr.ref,libhttpd.dll
+#)
 SET(install_targets ${install_targets} libhttpd)
 SET(install_bin_pdb ${install_bin_pdb} $<TARGET_PDB_FILE:libhttpd>)
 TARGET_LINK_LIBRARIES(libhttpd ${EXTRA_LIBS} ${APR_LIBRARIES} ${PCRE_LIBRARIES} ${HTTPD_SYSTEM_LIBS})
@@ -846,12 +880,12 @@ ENDIF()
 GET_PROPERTY(tmp_includes TARGET ab PROPERTY INCLUDE_DIRECTORIES)
 
 # getting duplicate manifest error with ApacheMonitor
-# ADD_EXECUTABLE(ApacheMonitor support/win32/ApacheMonitor.c support/win32/ApacheMonitor.rc)
-# SET(install_targets ${install_targets} ApacheMonitor)
-# SET(install_bin_pdb ${install_bin_pdb} $<TARGET_PDB_FILE:ApacheMonitor>)
-# SET_TARGET_PROPERTIES(ApacheMonitor PROPERTIES WIN32_EXECUTABLE TRUE)
-# SET_TARGET_PROPERTIES(ApacheMonitor PROPERTIES COMPILE_FLAGS "-DAPP_FILE -DLONG_NAME=ApacheMonitor -DBIN_NAME=ApacheMonitor.exe ${EXTRA_COMPILE_FLAGS}")
-# TARGET_LINK_LIBRARIES(ApacheMonitor ${EXTRA_LIBS} ${HTTPD_SYSTEM_LIBS} comctl32 wtsapi32)
+ADD_EXECUTABLE(ApacheMonitor support/win32/ApacheMonitor.c support/win32/ApacheMonitor.rc)
+SET(install_targets ${install_targets} ApacheMonitor)
+SET(install_bin_pdb ${install_bin_pdb} $<TARGET_PDB_FILE:ApacheMonitor>)
+SET_TARGET_PROPERTIES(ApacheMonitor PROPERTIES WIN32_EXECUTABLE TRUE)
+SET_TARGET_PROPERTIES(ApacheMonitor PROPERTIES COMPILE_FLAGS "-DAPP_FILE -DLONG_NAME=ApacheMonitor -DBIN_NAME=ApacheMonitor.exe ${EXTRA_COMPILE_FLAGS}")
+TARGET_LINK_LIBRARIES(ApacheMonitor ${EXTRA_LIBS} ${HTTPD_SYSTEM_LIBS} comctl32 wtsapi32)
 
 ###########  CONFIGURATION FILES ###########
 # Set up variables used in the .conf file templates
