 support/win32/ApacheMonitor.rc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/support/win32/ApacheMonitor.rc b/support/win32/ApacheMonitor.rc
index d4203bf926..9d8211e830 100644
--- a/support/win32/ApacheMonitor.rc
+++ b/support/win32/ApacheMonitor.rc
@@ -26,7 +26,7 @@
 IDI_APSRVMON            ICON    DISCARDABLE     "ApacheMonitor.ico"
 IDI_ICOSTOP             ICON    DISCARDABLE     "apstop.ico"
 IDI_ICORUN              ICON    DISCARDABLE     "aprun.ico"
-CREATEPROCESS_MANIFEST_RESOURCE_ID RT_MANIFEST  "ApacheMonitor.manifest"
+// CREATEPROCESS_MANIFEST_RESOURCE_ID RT_MANIFEST  "ApacheMonitor.manifest"
 
 IDD_DLGSERVICES DIALOGEX 0, 0, 350, 192
 STYLE DS_MODALFRAME | DS_SETFOREGROUND | WS_MINIMIZEBOX | WS_VISIBLE | 
