diff --git "a/c:\\temp\\TortoiseGit\\ApacheMonitor-d495914.000.rc" "b/C:\\src\\httpd\\support\\win32\\ApacheMonitor.rc"
index d4203bf926..6e519880ef 100644
--- "a/c:\\temp\\TortoiseGit\\ApacheMonitor-d495914.000.rc"
+++ "b/C:\\src\\httpd\\support\\win32\\ApacheMonitor.rc"
@@ -26,7 +26,7 @@
 IDI_APSRVMON            ICON    DISCARDABLE     "ApacheMonitor.ico"
 IDI_ICOSTOP             ICON    DISCARDABLE     "apstop.ico"
 IDI_ICORUN              ICON    DISCARDABLE     "aprun.ico"
-CREATEPROCESS_MANIFEST_RESOURCE_ID RT_MANIFEST  "ApacheMonitor.manifest"
+//CREATEPROCESS_MANIFEST_RESOURCE_ID RT_MANIFEST  "ApacheMonitor.manifest"
 
 IDD_DLGSERVICES DIALOGEX 0, 0, 350, 192
 STYLE DS_MODALFRAME | DS_SETFOREGROUND | WS_MINIMIZEBOX | WS_VISIBLE | 
