diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3cd8a34d14..dd6da43929 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,6 +26,7 @@ FIND_PACKAGE(LibXml2)
 FIND_PACKAGE(Lua51)
 FIND_PACKAGE(OpenSSL)
 FIND_PACKAGE(ZLIB)
+FIND_PACKAGE(CURL)
 
 # Options for support libraries not supported by cmake-bundled FindFOO
 
@@ -202,6 +203,7 @@ ENDIF()
 MESSAGE(STATUS "")
 MESSAGE(STATUS "Summary of feature detection:")
 MESSAGE(STATUS "")
+MESSAGE(STATUS "CURL_FOUND ............... : ${CURL_FOUND}")
 MESSAGE(STATUS "LIBXML2_FOUND ............ : ${LIBXML2_FOUND}")
 MESSAGE(STATUS "LUA51_FOUND .............. : ${LUA51_FOUND}")
 MESSAGE(STATUS "NGHTTP2_FOUND ............ : ${NGHTTP2_FOUND}")
@@ -351,6 +353,8 @@ SET(MODULE_LIST
   "modules/slotmem/mod_slotmem_plain+I+slotmem provider that uses plain memory"
   "modules/slotmem/mod_slotmem_shm+I+slotmem provider that uses shared memory"
   "modules/ssl/mod_ssl+i+SSL/TLS support"
+  "modules/md/mod_md+i+certificate provisioning via the ACME protocol"
+  "modules/fcgid/mod_fcgid+i+FastCGI implementation for Apache HTTP"
   "modules/test/mod_dialup+O+rate limits static files to dialup modem speeds"
   "modules/test/mod_optional_fn_export+O+example optional function exporter"
   "modules/test/mod_optional_fn_import+O+example optional function importer"
@@ -497,6 +501,38 @@ SET(mod_ssl_extra_sources
   modules/ssl/ssl_util.c             modules/ssl/ssl_util_ocsp.c
   modules/ssl/ssl_util_ssl.c         modules/ssl/ssl_util_stapling.c
 )
+IF(CURL_FOUND)
+  SET(mod_md_extra_includes           "${CURL_INCLUDE_DIR};${OPENSSL_INCLUDE_DIR};${JANSSON_INCLUDE_DIR}")
+  SET(mod_md_extra_libs               "${CURL_LIBRARIES};${OPENSSL_LIBRARIES};${JANSSON_LIBRARIES}")
+ENDIF()
+SET(mod_md_extra_sources
+modules/md/md_acme_authz.c
+modules/md/md_acme_acct.c
+modules/md/md_acme.c
+modules/md/md_core.c
+modules/md/md_store.c
+modules/md/md_curl.c
+modules/md/md_http.c
+modules/md/md_jws.c
+modules/md/mod_md_os.c
+modules/md/md_log.c
+modules/md/mod_md.c
+modules/md/md_crypt.c
+modules/md/md_acme_drive.c
+modules/md/md_util.c
+modules/md/md_reg.c
+modules/md/md_store_fs.c
+modules/md/mod_md_config.c
+modules/md/md_json.c
+)
+SET(mod_fcgid_extra_sources
+modules/fcgid/fcgid_proc_win.c		modules/fcgid/fcgid_proctbl_win.c
+modules/fcgid/fcgid_protocol.c		modules/fcgid/fcgid_spawn_ctl.c
+modules/fcgid/mod_fcgid.c		modules/fcgid/fcgid_bridge.c
+modules/fcgid/fcgid_bucket.c		modules/fcgid/fcgid_conf.c
+modules/fcgid/fcgid_filter.c		modules/fcgid/fcgid_pm_main.c
+modules/fcgid/fcgid_pm_win.c		
+)
 SET(mod_status_extra_defines         STATUS_DECLARE_EXPORT)
 SET(mod_watchdog_install_lib 1)
 SET(mod_xml2enc_requires             LIBXML2_FOUND)
@@ -564,7 +600,7 @@ ELSE()
 ENDIF()
 
 # for easy reference from .dll/.so builds
-CONFIGURE_FILE(os/win32/BaseAddr.ref ${PROJECT_BINARY_DIR}/ COPYONLY)
+# CONFIGURE_FILE(os/win32/BaseAddr.ref ${PROJECT_BINARY_DIR}/ COPYONLY)
 
 ADD_EXECUTABLE(gen_test_char server/gen_test_char.c)
 GET_TARGET_PROPERTY(GEN_TEST_CHAR_EXE gen_test_char LOCATION)
@@ -636,6 +672,8 @@ SET(LIBHTTPD_SOURCES
 
 CONFIGURE_FILE(os/win32/win32_config_layout.h
                ${PROJECT_BINARY_DIR}/ap_config_layout.h)
+CONFIGURE_FILE(server/mpm/winnt/mpm_winnt.h
+               ${PROJECT_BINARY_DIR}/mpm_winnt.h)
 
 SET(HTTPD_INCLUDE_DIRECTORIES
   ${PROJECT_BINARY_DIR}
@@ -651,6 +689,8 @@ SET(HTTPD_INCLUDE_DIRECTORIES
   ${CMAKE_CURRENT_SOURCE_DIR}/modules/proxy
   ${CMAKE_CURRENT_SOURCE_DIR}/modules/session
   ${CMAKE_CURRENT_SOURCE_DIR}/modules/ssl
+  ${CMAKE_CURRENT_SOURCE_DIR}/modules/md
+  ${CMAKE_CURRENT_SOURCE_DIR}/modules/fcgid
   ${CMAKE_CURRENT_SOURCE_DIR}/server
   ${APR_INCLUDE_DIR}
   ${PCRE_INCLUDE_DIR}
@@ -677,6 +717,7 @@ SET(other_installed_h
   ${CMAKE_CURRENT_SOURCE_DIR}/modules/session/mod_session.h
   ${CMAKE_CURRENT_SOURCE_DIR}/modules/ssl/mod_ssl.h
   ${CMAKE_CURRENT_SOURCE_DIR}/modules/ssl/mod_ssl_openssl.h
+  ${CMAKE_CURRENT_SOURCE_DIR}/modules/md/mod_md.h
 )
 # When mod_serf is buildable, don't forget to copy modules/proxy/mod_serf.h
 
@@ -765,7 +806,7 @@ FOREACH (mod ${MODULE_PATHS})
     SET(mod_extra_libs "${mod_name}_extra_libs")
     SET_TARGET_PROPERTIES(${mod_name} PROPERTIES
       SUFFIX .so
-      LINK_FLAGS /base:@${PROJECT_BINARY_DIR}/BaseAddr.ref,${mod_name}.so
+      # LINK_FLAGS /base:@${PROJECT_BINARY_DIR}/BaseAddr.ref,${mod_name}.so
     )
     TARGET_LINK_LIBRARIES(${mod_name} ${${mod_extra_libs}} libhttpd ${EXTRA_LIBS} ${APR_LIBRARIES} ${HTTPD_SYSTEM_LIBS})
     DEFINE_WITH_BLANKS(define_long_name "LONG_NAME" "${mod_name} for Apache HTTP Server")
@@ -790,9 +831,9 @@ ENDFOREACH()
 
 ###########   HTTPD LIBRARIES   ############
 ADD_LIBRARY(libhttpd SHARED ${LIBHTTPD_SOURCES} build/win32/httpd.rc)
-SET_TARGET_PROPERTIES(libhttpd PROPERTIES
-  LINK_FLAGS /base:@${PROJECT_BINARY_DIR}/BaseAddr.ref,libhttpd.dll
-)
+#SET_TARGET_PROPERTIES(libhttpd PROPERTIES
+#  LINK_FLAGS /base:@${PROJECT_BINARY_DIR}/BaseAddr.ref,libhttpd.dll
+#)
 SET(install_targets ${install_targets} libhttpd)
 SET(install_bin_pdb ${install_bin_pdb} $<TARGET_PDB_FILE:libhttpd>)
 TARGET_LINK_LIBRARIES(libhttpd ${EXTRA_LIBS} ${APR_LIBRARIES} ${PCRE_LIBRARIES} ${HTTPD_SYSTEM_LIBS})
@@ -846,12 +887,12 @@ ENDIF()
 GET_PROPERTY(tmp_includes TARGET ab PROPERTY INCLUDE_DIRECTORIES)
 
 # getting duplicate manifest error with ApacheMonitor
-# ADD_EXECUTABLE(ApacheMonitor support/win32/ApacheMonitor.c support/win32/ApacheMonitor.rc)
-# SET(install_targets ${install_targets} ApacheMonitor)
-# SET(install_bin_pdb ${install_bin_pdb} $<TARGET_PDB_FILE:ApacheMonitor>)
-# SET_TARGET_PROPERTIES(ApacheMonitor PROPERTIES WIN32_EXECUTABLE TRUE)
-# SET_TARGET_PROPERTIES(ApacheMonitor PROPERTIES COMPILE_FLAGS "-DAPP_FILE -DLONG_NAME=ApacheMonitor -DBIN_NAME=ApacheMonitor.exe ${EXTRA_COMPILE_FLAGS}")
-# TARGET_LINK_LIBRARIES(ApacheMonitor ${EXTRA_LIBS} ${HTTPD_SYSTEM_LIBS} comctl32 wtsapi32)
+ADD_EXECUTABLE(ApacheMonitor support/win32/ApacheMonitor.c support/win32/ApacheMonitor.rc)
+SET(install_targets ${install_targets} ApacheMonitor)
+SET(install_bin_pdb ${install_bin_pdb} $<TARGET_PDB_FILE:ApacheMonitor>)
+SET_TARGET_PROPERTIES(ApacheMonitor PROPERTIES WIN32_EXECUTABLE TRUE)
+SET_TARGET_PROPERTIES(ApacheMonitor PROPERTIES COMPILE_FLAGS "-DAPP_FILE -DLONG_NAME=ApacheMonitor -DBIN_NAME=ApacheMonitor.exe ${EXTRA_COMPILE_FLAGS}")
+TARGET_LINK_LIBRARIES(ApacheMonitor ${EXTRA_LIBS} ${HTTPD_SYSTEM_LIBS} comctl32 wtsapi32)
 
 ###########  CONFIGURATION FILES ###########
 # Set up variables used in the .conf file templates
@@ -953,6 +994,9 @@ MESSAGE(STATUS "  Brotli include directory......... : ${BROTLI_INCLUDE_DIR}")
 MESSAGE(STATUS "  Brotli libraries ................ : ${BROTLI_LIBRARIES}")
 MESSAGE(STATUS "  Extra include directories ....... : ${EXTRA_INCLUDES}")
 MESSAGE(STATUS "  Extra compile flags ............. : ${EXTRA_COMPILE_FLAGS}")
+MESSAGE(STATUS "  Extra module link flags ......... : ${CMAKE_MODULE_LINKER_FLAGS}")
+MESSAGE(STATUS "  Extra shared link flags ......... : ${CMAKE_SHARED_LINKER_FLAGS}")
+MESSAGE(STATUS "  Extra static link flags ......... : ${CMAKE_STATIC_LINKER_FLAGS}")
 MESSAGE(STATUS "  Extra libraries ................. : ${EXTRA_LIBS}")
 
 MESSAGE(STATUS "  Modules built and loaded:")
diff --git a/modules/proxy/mod_proxy_fcgi.c b/modules/proxy/mod_proxy_fcgi.c
index 41292e8726..9f7854f2a2 100644
--- a/modules/proxy/mod_proxy_fcgi.c
+++ b/modules/proxy/mod_proxy_fcgi.c
@@ -357,6 +357,10 @@ static apr_status_t send_environment(proxy_conn_rec *conn, request_rec *r,
 
         if (newfname) {
             newfname = ap_strchr(newfname, '/');
+			// Windows
+			if ((ap_strstr(newfname, ":/")) != NULL)
+				memmove(newfname, newfname+1, strlen(newfname));
+			ap_log_rerror(APLOG_MARK, APLOG_DEBUG, 0, r, APLOGNO(01078) "SCRIPT_FILENAME %s", newfname);
             r->filename = newfname;
         }
     }
diff --git a/server/config.c b/server/config.c
index ffd26fd592..ce05035a2d 100644
--- a/server/config.c
+++ b/server/config.c
@@ -158,6 +158,17 @@ AP_DECLARE(int) ap_run_post_config(apr_pool_t *pconf,
 }
  */
 
+ AP_DECLARE(int) ap_get_module_flags(const module *m)
+{
+    if (m->version < AP_MODULE_FLAGS_MMN_MAJOR
+            || (m->version == AP_MODULE_FLAGS_MMN_MAJOR
+                && (m->minor_version < AP_MODULE_FLAGS_MMN_MINOR))) {
+        return 0;
+    }
+
+    return m->flags;
+}
+
 AP_IMPLEMENT_HOOK_RUN_ALL(int, open_logs,
                           (apr_pool_t *pconf, apr_pool_t *plog,
                            apr_pool_t *ptemp, server_rec *s),
diff --git a/server/main.c b/server/main.c
index 3a4b1a6254..0e99be954a 100644
--- a/server/main.c
+++ b/server/main.c
@@ -44,6 +44,7 @@
 #if APR_HAVE_UNISTD_H
 #include <unistd.h>
 #endif
+#include "openssl\applink.c"
 
 /* WARNING: Win32 binds http_main.c dynamically to the server. Please place
  *          extern functions and global data in another appropriate module.
diff --git a/support/win32/ApacheMonitor.rc b/support/win32/ApacheMonitor.rc
index d4203bf926..9d8211e830 100644
--- a/support/win32/ApacheMonitor.rc
+++ b/support/win32/ApacheMonitor.rc
@@ -26,7 +26,7 @@
 IDI_APSRVMON            ICON    DISCARDABLE     "ApacheMonitor.ico"
 IDI_ICOSTOP             ICON    DISCARDABLE     "apstop.ico"
 IDI_ICORUN              ICON    DISCARDABLE     "aprun.ico"
-CREATEPROCESS_MANIFEST_RESOURCE_ID RT_MANIFEST  "ApacheMonitor.manifest"
+// CREATEPROCESS_MANIFEST_RESOURCE_ID RT_MANIFEST  "ApacheMonitor.manifest"
 
 IDD_DLGSERVICES DIALOGEX 0, 0, 350, 192
 STYLE DS_MODALFRAME | DS_SETFOREGROUND | WS_MINIMIZEBOX | WS_VISIBLE | 
