 server/main.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/server/main.c b/server/main.c
index ec0ea68d9b..cffb65b05f 100644
--- a/server/main.c
+++ b/server/main.c
@@ -44,6 +44,7 @@
 #if APR_HAVE_UNISTD_H
 #include <unistd.h>
 #endif
+#include "openssl\applink.c"
 
 /* WARNING: Win32 binds http_main.c dynamically to the server. Please place
  *          extern functions and global data in another appropriate module.
