diff --git "a/c:\\temp\\TortoiseGit\\mod_proxy_fcgi-22efe86.000.c" "b/C:\\src\\httpd\\modules\\proxy\\mod_proxy_fcgi.c"
index 41292e8726..9f7854f2a2 100644
--- "a/c:\\temp\\TortoiseGit\\mod_proxy_fcgi-22efe86.000.c"
+++ "b/C:\\src\\httpd\\modules\\proxy\\mod_proxy_fcgi.c"
@@ -357,6 +357,10 @@ static apr_status_t send_environment(proxy_conn_rec *conn, request_rec *r,
 
         if (newfname) {
             newfname = ap_strchr(newfname, '/');
+			// Windows
+			if ((ap_strstr(newfname, ":/")) != NULL)
+				memmove(newfname, newfname+1, strlen(newfname));
+			ap_log_rerror(APLOG_MARK, APLOG_DEBUG, 0, r, APLOGNO(01078) "SCRIPT_FILENAME %s", newfname);
             r->filename = newfname;
         }
     }
